// @mantine/core/cjs/components/Button/ButtonGroup/ButtonGroup.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../../core/utils/units-converters/rem.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Button_module = require("../Button.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Button/ButtonGroup/ButtonGroup.tsx
const defaultProps = { orientation: "horizontal" };
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { borderWidth }) => ({ group: { "--button-border-width": require_rem.rem(borderWidth) } }));
const ButtonGroup = require_factory.factory((_props) => {
	const props = require_use_props.useProps("ButtonGroup", defaultProps, _props);
	const { className, style, classNames, styles, unstyled, orientation, vars, borderWidth, mod, attributes, ...others } = require_use_props.useProps("ButtonGroup", defaultProps, _props);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...require_use_styles.useStyles({
			name: "ButtonGroup",
			props,
			classes: require_Button_module.default,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver,
			rootSelector: "group"
		})("group"),
		mod: [{ "data-orientation": orientation }, mod],
		role: "group",
		...others
	});
});
ButtonGroup.classes = require_Button_module.default;
ButtonGroup.varsResolver = varsResolver;
ButtonGroup.displayName = "@mantine/core/ButtonGroup";
//#endregion
exports.ButtonGroup = ButtonGroup;
