// @mantine/core/cjs/components/NativeSelect/NativeSelect.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_InputBase = require("../InputBase/InputBase.cjs");
const require_get_parsed_combobox_data = require("../Combobox/get-parsed-combobox-data/get-parsed-combobox-data.cjs");
const require_ComboboxChevron = require("../Combobox/ComboboxChevron/ComboboxChevron.cjs");
const require_NativeSelectOption = require("./NativeSelectOption.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/NativeSelect/NativeSelect.tsx
const defaultProps = {
	size: "sm",
	rightSectionPointerEvents: "none"
};
const NativeSelect = require_factory.factory((props) => {
	const { data, children, size, error, rightSection, unstyled, ...others } = require_use_props.useProps([
		"Input",
		"InputWrapper",
		"NativeSelect"
	], defaultProps, props);
	const options = require_get_parsed_combobox_data.getParsedComboboxData(data).map((item, index) => /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_NativeSelectOption.NativeSelectOption, { data: item }, index));
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_InputBase.InputBase, {
		component: "select",
		...others,
		__staticSelector: "NativeSelect",
		size,
		pointer: true,
		error,
		unstyled,
		rightSection: rightSection || /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_ComboboxChevron.ComboboxChevron, {
			size,
			error,
			unstyled
		}),
		children: children || options
	});
});
NativeSelect.classes = require_InputBase.InputBase.classes;
NativeSelect.displayName = "@mantine/core/NativeSelect";
//#endregion
exports.NativeSelect = NativeSelect;
