// @mantine/core/cjs/components/HoverCard/HoverCard.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_Popover = require("../Popover/Popover.cjs");
const require_HoverCard_context = require("./HoverCard.context.cjs");
const require_HoverCardGroup = require("./HoverCardGroup/HoverCardGroup.cjs");
const require_HoverCardDropdown = require("./HoverCardDropdown/HoverCardDropdown.cjs");
const require_HoverCardTarget = require("./HoverCardTarget/HoverCardTarget.cjs");
const require_use_hover_card = require("./use-hover-card.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/HoverCard/HoverCard.tsx
const defaultProps = {
	openDelay: 0,
	closeDelay: 150,
	initiallyOpened: false
};
function HoverCard(props) {
	const { children, onOpen, onClose, openDelay, closeDelay, initiallyOpened, ...others } = require_use_props.useProps("HoverCard", defaultProps, props);
	const hoverCard = require_use_hover_card.useHoverCard({
		openDelay,
		closeDelay,
		defaultOpened: initiallyOpened,
		onOpen,
		onClose
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_HoverCard_context.HoverCardContext, {
		value: {
			openDropdown: hoverCard.openDropdown,
			closeDropdown: hoverCard.closeDropdown,
			getReferenceProps: hoverCard.getReferenceProps,
			getFloatingProps: hoverCard.getFloatingProps,
			reference: hoverCard.reference,
			floating: hoverCard.floating
		},
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Popover.Popover, {
			...others,
			opened: hoverCard.opened,
			__staticSelector: "HoverCard",
			children
		})
	});
}
HoverCard.displayName = "@mantine/core/HoverCard";
HoverCard.Target = require_HoverCardTarget.HoverCardTarget;
HoverCard.Dropdown = require_HoverCardDropdown.HoverCardDropdown;
HoverCard.Group = require_HoverCardGroup.HoverCardGroup;
HoverCard.extend = (input) => input;
//#endregion
exports.HoverCard = HoverCard;
