// @mantine/core/cjs/components/Skeleton/Skeleton.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../core/utils/units-converters/rem.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Skeleton_module = require("./Skeleton.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Skeleton/Skeleton.tsx
const defaultProps = {
	visible: true,
	animate: true
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { width, height, radius, circle }) => ({ root: {
	"--skeleton-height": require_rem.rem(height),
	"--skeleton-width": circle ? require_rem.rem(height) : require_rem.rem(width),
	"--skeleton-radius": circle ? "1000px" : radius === void 0 ? void 0 : require_get_size.getRadius(radius)
} }));
const Skeleton = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Skeleton", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, width, height, circle, visible, radius, animate, mod, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...require_use_styles.useStyles({
			name: "Skeleton",
			classes: require_Skeleton_module.default,
			props,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver
		})("root"),
		mod: [{
			visible,
			animate
		}, mod],
		...others
	});
});
Skeleton.classes = require_Skeleton_module.default;
Skeleton.varsResolver = varsResolver;
Skeleton.displayName = "@mantine/core/Skeleton";
//#endregion
exports.Skeleton = Skeleton;
