// @mantine/core/cjs/components/Combobox/ComboboxChevron/ComboboxChevron.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Combobox_module = require("../Combobox.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Combobox/ComboboxChevron/ComboboxChevron.tsx
const defaultProps = { error: null };
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { size, color }) => ({ chevron: {
	"--combobox-chevron-size": require_get_size.getSize(size, "combobox-chevron-size"),
	"--combobox-chevron-color": color ? require_get_theme_color.getThemeColor(color, theme) : void 0
} }));
const ComboboxChevron = require_factory.factory((_props) => {
	const props = require_use_props.useProps("ComboboxChevron", defaultProps, _props);
	const { size, error, style, className, classNames, styles, unstyled, vars, attributes, mod, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "ComboboxChevron",
		classes: require_Combobox_module.default,
		props,
		style,
		className,
		classNames,
		styles,
		unstyled,
		vars,
		varsResolver,
		attributes,
		rootSelector: "chevron"
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		component: "svg",
		...others,
		...getStyles("chevron"),
		size,
		viewBox: "0 0 15 15",
		fill: "none",
		xmlns: "http://www.w3.org/2000/svg",
		mod: [
			"combobox-chevron",
			{ error },
			mod
		],
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)("path", {
			d: "M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z",
			fill: "currentColor",
			fillRule: "evenodd",
			clipRule: "evenodd"
		})
	});
});
ComboboxChevron.classes = require_Combobox_module.default;
ComboboxChevron.varsResolver = varsResolver;
ComboboxChevron.displayName = "@mantine/core/ComboboxChevron";
//#endregion
exports.ComboboxChevron = ComboboxChevron;
