// @mantine/core/cjs/components/Tooltip/Tooltip.cjs
"use client";
const require_runtime = require("../../_virtual/_rolldown/runtime.cjs");
const require_get_default_z_index = require("../../core/utils/get-default-z-index/get-default-z-index.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_get_ref_prop = require("../../core/utils/get-ref-prop/get-ref-prop.cjs");
const require_get_single_element_child = require("../../core/utils/get-single-element-child/get-single-element-child.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_DirectionProvider = require("../../core/DirectionProvider/DirectionProvider.cjs");
const require_FloatingArrow = require("../../utils/Floating/FloatingArrow/FloatingArrow.cjs");
const require_get_floating_position = require("../../utils/Floating/get-floating-position/get-floating-position.cjs");
const require_OptionalPortal = require("../Portal/OptionalPortal.cjs");
const require_Transition = require("../Transition/Transition.cjs");
const require_get_transition_props = require("../Transition/get-transition-props/get-transition-props.cjs");
const require_Tooltip_module = require("./Tooltip.module.cjs");
const require_TooltipFloating = require("./TooltipFloating/TooltipFloating.cjs");
const require_TooltipGroup = require("./TooltipGroup/TooltipGroup.cjs");
const require_use_tooltip = require("./use-tooltip.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let clsx = require("clsx");
clsx = require_runtime.__toESM(clsx);
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Tooltip/Tooltip.tsx
const defaultProps = {
	position: "top",
	refProp: "ref",
	withinPortal: true,
	arrowSize: 4,
	arrowOffset: 5,
	arrowRadius: 0,
	arrowPosition: "side",
	offset: 5,
	transitionProps: {
		duration: 100,
		transition: "fade"
	},
	events: {
		hover: true,
		focus: false,
		touch: false
	},
	zIndex: require_get_default_z_index.getDefaultZIndex("popover"),
	middlewares: {
		flip: true,
		shift: true,
		inline: false
	}
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, variant, autoContrast }) => {
	const colors = theme.variantColorResolver({
		theme,
		color: color || theme.primaryColor,
		autoContrast,
		variant: variant || "filled"
	});
	return { tooltip: {
		"--tooltip-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--tooltip-bg": color ? colors.background : void 0,
		"--tooltip-color": color ? colors.color : void 0
	} };
});
const Tooltip = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Tooltip", defaultProps, _props);
	const { children, position, refProp, label, openDelay, closeDelay, onPositionChange, opened, defaultOpened, withinPortal, radius, color, classNames, styles, unstyled, style, className, withArrow, arrowSize, arrowOffset, arrowRadius, arrowPosition, offset, transitionProps, multiline, events, zIndex, disabled, onClick, onMouseEnter, onMouseLeave, inline, variant, keepMounted, vars, portalProps, mod, floatingStrategy, middlewares, autoContrast, attributes, target, ref, ...others } = props;
	const { dir } = require_DirectionProvider.useDirection();
	const arrowRef = (0, react.useRef)(null);
	const tooltip = require_use_tooltip.useTooltip({
		position: require_get_floating_position.getFloatingPosition(dir, position),
		closeDelay,
		openDelay,
		onPositionChange,
		opened,
		defaultOpened,
		events,
		arrowRef,
		arrowOffset,
		offset: typeof offset === "number" ? offset + (withArrow ? arrowSize / 2 : 0) : offset,
		inline,
		strategy: floatingStrategy,
		middlewares
	});
	(0, react.useEffect)(() => {
		const targetNode = target instanceof HTMLElement ? target : typeof target === "string" ? document.querySelector(target) : target?.current || null;
		if (targetNode) tooltip.reference(targetNode);
	}, [target, tooltip]);
	const getStyles = require_use_styles.useStyles({
		name: "Tooltip",
		props,
		classes: require_Tooltip_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		rootSelector: "tooltip",
		vars,
		varsResolver
	});
	const child = require_get_single_element_child.getSingleElementChild(children);
	if (!target && !child) throw new Error("[@mantine/core] Tooltip component children should be an element or a component that accepts ref, fragments, strings, numbers and other primitive values are not supported");
	const tooltipStyles = getStyles("tooltip");
	if (target) {
		const transition = require_get_transition_props.getTransitionProps(transitionProps, {
			duration: 100,
			transition: "fade"
		});
		return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(react_jsx_runtime.Fragment, { children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_OptionalPortal.OptionalPortal, {
			...portalProps,
			withinPortal,
			children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Transition.Transition, {
				...transition,
				keepMounted,
				mounted: !disabled && !!tooltip.opened,
				duration: tooltip.isGroupPhase ? 10 : transition.duration,
				children: (transitionStyles) => /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
					...others,
					"data-fixed": floatingStrategy === "fixed" || void 0,
					variant,
					mod: [{ multiline }, mod],
					...tooltipStyles,
					...tooltip.getFloatingProps({
						ref: tooltip.floating,
						className: tooltipStyles.className,
						style: {
							...tooltipStyles.style,
							...transitionStyles,
							zIndex,
							top: tooltip.y ?? 0,
							left: tooltip.x ?? 0
						}
					}),
					children: [label, /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_FloatingArrow.FloatingArrow, {
						ref: arrowRef,
						arrowX: tooltip.arrowX,
						arrowY: tooltip.arrowY,
						visible: withArrow,
						position: tooltip.placement,
						arrowSize,
						arrowOffset,
						arrowRadius,
						arrowPosition,
						...getStyles("arrow")
					})]
				})
			})
		}) });
	}
	const childProps = child.props;
	const targetRef = (0, _mantine_hooks.useMergedRef)(tooltip.reference, require_get_ref_prop.getRefProp(child), ref);
	const transition = require_get_transition_props.getTransitionProps(transitionProps, {
		duration: 100,
		transition: "fade"
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(react_jsx_runtime.Fragment, { children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_OptionalPortal.OptionalPortal, {
		...portalProps,
		withinPortal,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Transition.Transition, {
			...transition,
			keepMounted,
			mounted: !disabled && !!tooltip.opened,
			duration: tooltip.isGroupPhase ? 10 : transition.duration,
			children: (transitionStyles) => /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
				...others,
				"data-fixed": floatingStrategy === "fixed" || void 0,
				variant,
				mod: [{ multiline }, mod],
				...tooltip.getFloatingProps({
					ref: tooltip.floating,
					className: getStyles("tooltip").className,
					style: {
						...getStyles("tooltip").style,
						...transitionStyles,
						zIndex,
						top: tooltip.y ?? 0,
						left: tooltip.x ?? 0
					}
				}),
				children: [label, /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_FloatingArrow.FloatingArrow, {
					ref: arrowRef,
					arrowX: tooltip.arrowX,
					arrowY: tooltip.arrowY,
					visible: withArrow,
					position: tooltip.placement,
					arrowSize,
					arrowOffset,
					arrowRadius,
					arrowPosition,
					...getStyles("arrow")
				})]
			})
		})
	}), (0, react.cloneElement)(child, tooltip.getReferenceProps({
		onClick,
		onMouseEnter,
		onMouseLeave,
		onMouseMove: props.onMouseMove,
		onPointerDown: props.onPointerDown,
		onPointerEnter: props.onPointerEnter,
		...childProps,
		className: (0, clsx.default)(className, childProps.className),
		[refProp]: targetRef
	}))] });
});
Tooltip.classes = require_Tooltip_module.default;
Tooltip.varsResolver = varsResolver;
Tooltip.displayName = "@mantine/core/Tooltip";
Tooltip.Floating = require_TooltipFloating.TooltipFloating;
Tooltip.Group = require_TooltipGroup.TooltipGroup;
//#endregion
exports.Tooltip = Tooltip;
