// @mantine/core/esm/components/TagsInput/TagsInput.mjs
"use client";
import { useProps } from "../../core/MantineProvider/use-props/use-props.mjs";
import { useResolvedStylesApi } from "../../core/styles-api/use-resolved-styles-api/use-resolved-styles-api.mjs";
import { useStyles } from "../../core/styles-api/use-styles/use-styles.mjs";
import { extractStyleProps } from "../../core/Box/style-props/extract-style-props/extract-style-props.mjs";
import { factory } from "../../core/factory/factory.mjs";
import { InputBase } from "../InputBase/InputBase.mjs";
import { getParsedComboboxData } from "../Combobox/get-parsed-combobox-data/get-parsed-combobox-data.mjs";
import { getOptionsLockup } from "../Combobox/get-options-lockup/get-options-lockup.mjs";
import { useCombobox } from "../Combobox/use-combobox/use-combobox.mjs";
import { Combobox } from "../Combobox/Combobox.mjs";
import { OptionsDropdown } from "../Combobox/OptionsDropdown/OptionsDropdown.mjs";
import { usePillsReorder } from "../Combobox/use-pills-reorder/use-pills-reorder.mjs";
import { Pill } from "../Pill/Pill.mjs";
import { PillsInput } from "../PillsInput/PillsInput.mjs";
import { filterPickedTags } from "./filter-picked-tags.mjs";
import { getSplittedTags } from "./get-splitted-tags.mjs";
import { Fragment, useEffect, useRef } from "react";
import { useId as useId$1, useMergedRef, useUncontrolled } from "@mantine/hooks";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
//#region packages/@mantine/core/src/components/TagsInput/TagsInput.tsx
const defaultProps = {
	maxTags: Infinity,
	acceptValueOnBlur: true,
	splitChars: [","],
	hiddenInputValuesDivider: ",",
	openOnFocus: true,
	size: "sm"
};
const TagsInput = factory((_props) => {
	const props = useProps([
		"Input",
		"InputWrapper",
		"TagsInput"
	], defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, size, value, defaultValue, onChange, onKeyDown, maxTags, allowDuplicates, onDuplicate, variant, data, dropdownOpened, defaultDropdownOpened, onDropdownOpen, onDropdownClose, selectFirstOptionOnChange, selectFirstOptionOnDropdownOpen, onOptionSubmit, comboboxProps, filter, limit, withScrollArea, maxDropdownHeight, searchValue, defaultSearchValue, onSearchChange, readOnly, disabled, splitChars, onFocus, onBlur, onPaste, radius, rightSection, rightSectionWidth, rightSectionPointerEvents, rightSectionProps, leftSection, leftSectionWidth, leftSectionPointerEvents, leftSectionProps, inputContainer, inputWrapperOrder, withAsterisk, required, labelProps, descriptionProps, errorProps, wrapperProps, description, label, error, withErrorStyles, name, form, id, clearable, clearSectionMode, clearButtonProps, hiddenInputProps, hiddenInputValuesDivider, mod, renderOption, renderPill, onRemove, onClear, onMaxTags, scrollAreaProps, acceptValueOnBlur, isDuplicate, openOnFocus, attributes, ref, loading, loadingPosition, withPillsReorder, ...others } = props;
	const _id = useId$1(id);
	const parsedData = getParsedComboboxData(data);
	const optionsLockup = getOptionsLockup(parsedData);
	const inputRef = useRef(null);
	const _ref = useMergedRef(inputRef, ref);
	const combobox = useCombobox({
		opened: dropdownOpened,
		defaultOpened: defaultDropdownOpened,
		onDropdownOpen: () => {
			onDropdownOpen?.();
			if (selectFirstOptionOnDropdownOpen) combobox.selectFirstOption();
		},
		onDropdownClose: () => {
			onDropdownClose?.();
			combobox.resetSelectedOption();
		}
	});
	const { styleProps, rest: { type, autoComplete, ...rest } } = extractStyleProps(others);
	const [_value, setValue] = useUncontrolled({
		value,
		defaultValue,
		finalValue: [],
		onChange
	});
	const { getPillProps, getListProps, handleInputKeyDown } = usePillsReorder({
		value: _value,
		onChange: setValue,
		enabled: withPillsReorder && !disabled && !readOnly
	});
	const [_searchValue, setSearchValue] = useUncontrolled({
		value: searchValue,
		defaultValue: defaultSearchValue,
		finalValue: "",
		onChange: onSearchChange
	});
	const handleSearchChange = (value) => {
		setSearchValue(value);
		combobox.resetSelectedOption();
	};
	const getStyles = useStyles({
		name: "TagsInput",
		classes: {},
		props,
		classNames,
		styles,
		unstyled
	});
	const { resolvedClassNames, resolvedStyles } = useResolvedStylesApi({
		props,
		styles,
		classNames
	});
	const handleValueSelect = (val) => {
		if (isDuplicate ? isDuplicate(val, _value) : _value.some((tag) => tag.toLowerCase() === val.toLowerCase())) {
			onDuplicate?.(val);
			if (!allowDuplicates) {
				handleSearchChange("");
				return;
			}
		}
		if (_value.length >= maxTags) {
			onMaxTags?.(val);
			return;
		}
		onOptionSubmit?.(val);
		handleSearchChange("");
		if (val.length > 0) setValue([..._value, val]);
	};
	const handleInputKeydown = (event) => {
		onKeyDown?.(event);
		if (event.defaultPrevented || event.isPropagationStopped()) return;
		const inputValue = _searchValue.trim();
		const { length } = inputValue;
		if (splitChars.includes(event.key) && length > 0) {
			setValue(getSplittedTags({
				splitChars,
				allowDuplicates,
				maxTags,
				value: _searchValue,
				currentTags: _value,
				isDuplicate,
				onDuplicate
			}));
			handleSearchChange("");
			event.preventDefault();
		}
		if (event.key === "Enter" && length > 0 && !event.nativeEvent.isComposing) {
			event.preventDefault();
			if (!!document.querySelector(`#${combobox.listId} [data-combobox-option][data-combobox-selected]`)) return;
			handleValueSelect(inputValue);
		}
		if (event.key === "Backspace" && length === 0 && _value.length > 0 && !event.nativeEvent.isComposing && !readOnly) {
			onRemove?.(_value[_value.length - 1]);
			setValue(_value.slice(0, _value.length - 1));
		}
		handleInputKeyDown(event);
	};
	const handlePaste = (event) => {
		onPaste?.(event);
		event.preventDefault();
		if (event.clipboardData) {
			setValue(getSplittedTags({
				splitChars,
				allowDuplicates,
				maxTags,
				value: `${_searchValue}${event.clipboardData.getData("text/plain")}`,
				currentTags: _value,
				isDuplicate,
				onDuplicate
			}));
			handleSearchChange("");
		}
	};
	const values = _value.map((item, index) => {
		const onRemoveItem = () => {
			const next_value = _value.slice();
			next_value.splice(index, 1);
			setValue(next_value);
			onRemove?.(item);
		};
		const reorderProps = getPillProps(index);
		if (renderPill) return /* @__PURE__ */ jsx(Fragment, { children: renderPill({
			option: optionsLockup[item] || {
				value: item,
				label: item,
				disabled: false
			},
			value: item,
			onRemove: onRemoveItem,
			disabled: disabled || readOnly,
			reorderProps
		}) }, `${item}-${index}`);
		return /* @__PURE__ */ jsx(Pill, {
			withRemoveButton: !readOnly,
			onRemove: onRemoveItem,
			unstyled,
			disabled,
			attributes,
			...getStyles("pill"),
			...reorderProps,
			children: item
		}, `${item}-${index}`);
	});
	useEffect(() => {
		if (selectFirstOptionOnChange) combobox.selectFirstOption();
	}, [
		selectFirstOptionOnChange,
		_value,
		_searchValue
	]);
	const clearButton = /* @__PURE__ */ jsx(Combobox.ClearButton, {
		...clearButtonProps,
		onClear: () => {
			setValue([]);
			handleSearchChange("");
			inputRef.current?.focus();
			combobox.openDropdown();
			onClear?.();
		}
	});
	return /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsxs(Combobox, {
		store: combobox,
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		unstyled,
		size,
		readOnly,
		__staticSelector: "TagsInput",
		attributes,
		onOptionSubmit: (val) => {
			onOptionSubmit?.(val);
			handleSearchChange("");
			if (_value.length >= maxTags) onMaxTags?.(val);
			else setValue([..._value, optionsLockup[val].value]);
			combobox.resetSelectedOption();
		},
		...comboboxProps,
		children: [/* @__PURE__ */ jsx(Combobox.DropdownTarget, { children: /* @__PURE__ */ jsx(PillsInput, {
			...styleProps,
			__staticSelector: "TagsInput",
			classNames: resolvedClassNames,
			styles: resolvedStyles,
			unstyled,
			size,
			className,
			style,
			variant,
			disabled,
			radius,
			rightSection,
			__clearSection: clearButton,
			__clearable: clearable && _value.length > 0 && !disabled && !readOnly,
			__clearSectionMode: clearSectionMode,
			rightSectionWidth,
			rightSectionPointerEvents,
			rightSectionProps,
			leftSection,
			leftSectionWidth,
			leftSectionPointerEvents,
			leftSectionProps,
			loading,
			loadingPosition,
			inputContainer,
			inputWrapperOrder,
			withAsterisk,
			required,
			labelProps,
			descriptionProps,
			errorProps,
			wrapperProps,
			description,
			label,
			error,
			withErrorStyles,
			__stylesApiProps: {
				...props,
				multiline: true
			},
			id: _id,
			mod,
			attributes,
			children: /* @__PURE__ */ jsxs(Pill.Group, {
				disabled,
				unstyled,
				...getStyles("pillsList"),
				...getListProps(),
				children: [values, /* @__PURE__ */ jsx(Combobox.EventsTarget, {
					autoComplete,
					withExpandedAttribute: true,
					children: /* @__PURE__ */ jsx(PillsInput.Field, {
						...rest,
						ref: _ref,
						...getStyles("inputField"),
						unstyled,
						onKeyDown: handleInputKeydown,
						onFocus: (event) => {
							onFocus?.(event);
							openOnFocus && combobox.openDropdown();
						},
						onBlur: (event) => {
							onBlur?.(event);
							acceptValueOnBlur && handleValueSelect(_searchValue);
							combobox.closeDropdown();
						},
						onPaste: handlePaste,
						value: _searchValue,
						onChange: (event) => handleSearchChange(event.currentTarget.value),
						required: required && _value.length === 0,
						disabled,
						readOnly,
						id: _id
					})
				})]
			})
		}) }), /* @__PURE__ */ jsx(OptionsDropdown, {
			data: filterPickedTags({
				data: parsedData,
				value: _value
			}),
			hidden: readOnly || disabled,
			filter,
			search: _searchValue,
			limit,
			hiddenWhenEmpty: true,
			withScrollArea,
			maxDropdownHeight,
			unstyled,
			labelId: label ? `${_id}-label` : void 0,
			"aria-label": label ? void 0 : others["aria-label"],
			renderOption,
			scrollAreaProps
		})]
	}), /* @__PURE__ */ jsx(Combobox.HiddenInput, {
		name,
		form,
		value: _value,
		valuesDivider: hiddenInputValuesDivider,
		disabled,
		...hiddenInputProps
	})] });
});
TagsInput.classes = {
	...InputBase.classes,
	...Combobox.classes
};
TagsInput.displayName = "@mantine/core/TagsInput";
//#endregion
export { TagsInput };
