// @mantine/core/cjs/components/Code/Code.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Code_module = require("./Code.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Code/Code.tsx
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { color }) => ({ root: { "--code-bg": color ? require_get_theme_color.getThemeColor(color, theme) : void 0 } }));
const Code = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Code", null, _props);
	const { classNames, className, style, styles, unstyled, vars, color, block, mod, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Code",
		props,
		classes: require_Code_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		component: block ? "pre" : "code",
		mod: [{ block }, mod],
		...getStyles("root"),
		...others,
		dir: "ltr"
	});
});
Code.classes = require_Code_module.default;
Code.varsResolver = varsResolver;
Code.displayName = "@mantine/core/Code";
//#endregion
exports.Code = Code;
