// @mantine/core/cjs/components/Tabs/Tabs.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_safe_id = require("../../core/utils/get-safe-id/get-safe-id.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_get_contrast_color = require("../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs");
const require_get_auto_contrast_value = require("../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Tabs_context = require("./Tabs.context.cjs");
const require_Tabs_module = require("./Tabs.module.cjs");
const require_TabsList = require("./TabsList/TabsList.cjs");
const require_TabsPanel = require("./TabsPanel/TabsPanel.cjs");
const require_TabsTab = require("./TabsTab/TabsTab.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Tabs/Tabs.tsx
const VALUE_ERROR = "Tabs.Tab or Tabs.Panel component was rendered with invalid value or without value";
const defaultProps = {
	keepMounted: true,
	keepMountedMode: "activity",
	orientation: "horizontal",
	loop: true,
	activateTabWithKeyboard: true,
	variant: "default",
	placement: "left"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, autoContrast }) => ({ root: {
	"--tabs-radius": require_get_size.getRadius(radius),
	"--tabs-color": require_get_theme_color.getThemeColor(color, theme),
	"--tabs-text-color": require_get_auto_contrast_value.getAutoContrastValue(autoContrast, theme) ? require_get_contrast_color.getContrastColor({
		color,
		theme,
		autoContrast
	}) : void 0
} }));
const Tabs = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Tabs", defaultProps, _props);
	const { defaultValue, value, onChange, orientation, children, loop, id, activateTabWithKeyboard, allowTabDeactivation, variant, color, radius, inverted, placement, keepMounted, keepMountedMode, classNames, styles, unstyled, className, style, vars, autoContrast, mod, attributes, ...others } = props;
	const uid = (0, _mantine_hooks.useId)(id);
	const [currentTab, setCurrentTab] = (0, _mantine_hooks.useUncontrolled)({
		value,
		defaultValue,
		finalValue: null,
		onChange
	});
	const getStyles = require_use_styles.useStyles({
		name: "Tabs",
		props,
		classes: require_Tabs_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Tabs_context.TabsProvider, {
		value: {
			placement,
			value: currentTab,
			orientation,
			id: uid,
			loop,
			activateTabWithKeyboard,
			getTabId: require_get_safe_id.getSafeId(`${uid}-tab`, VALUE_ERROR),
			getPanelId: require_get_safe_id.getSafeId(`${uid}-panel`, VALUE_ERROR),
			onChange: setCurrentTab,
			allowTabDeactivation,
			variant,
			color,
			radius,
			inverted,
			keepMounted,
			keepMountedMode,
			unstyled,
			getStyles
		},
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			id: uid,
			variant,
			mod: [{
				orientation,
				inverted: orientation === "horizontal" && inverted,
				placement: orientation === "vertical" && placement
			}, mod],
			...getStyles("root"),
			...others,
			children
		})
	});
});
Tabs.classes = require_Tabs_module.default;
Tabs.varsResolver = varsResolver;
Tabs.displayName = "@mantine/core/Tabs";
Tabs.Tab = require_TabsTab.TabsTab;
Tabs.Panel = require_TabsPanel.TabsPanel;
Tabs.List = require_TabsList.TabsList;
//#endregion
exports.Tabs = Tabs;
