// @mantine/core/cjs/components/Input/InputLabel/InputLabel.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_InputWrapper_context = require("../InputWrapper.context.cjs");
const require_Input_module = require("../Input.module.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Input/InputLabel/InputLabel.tsx
const defaultProps = { labelElement: "label" };
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { size }) => ({ label: {
	"--input-label-size": require_get_size.getFontSize(size),
	"--input-asterisk-color": void 0
} }));
const InputLabel = require_factory.factory((_props) => {
	const props = require_use_props.useProps("InputLabel", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, labelElement, required, htmlFor, onMouseDown, children, __staticSelector, mod, attributes, ...others } = props;
	const _getStyles = require_use_styles.useStyles({
		name: ["InputWrapper", __staticSelector],
		props,
		classes: require_Input_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		rootSelector: "label",
		vars,
		varsResolver
	});
	const ctx = (0, react.use)(require_InputWrapper_context.InputWrapperContext);
	const getStyles = ctx?.getStyles || _getStyles;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		...getStyles("label", ctx?.getStyles ? {
			className,
			style
		} : void 0),
		component: labelElement,
		htmlFor: labelElement === "label" ? htmlFor : void 0,
		mod: [{ required }, mod],
		onMouseDown: (event) => {
			onMouseDown?.(event);
			if (!event.defaultPrevented && event.detail > 1) event.preventDefault();
		},
		...others,
		children: [children, required && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
			...getStyles("required"),
			"aria-hidden": true,
			children: " *"
		})]
	});
});
InputLabel.classes = require_Input_module.default;
InputLabel.varsResolver = varsResolver;
InputLabel.displayName = "@mantine/core/InputLabel";
//#endregion
exports.InputLabel = InputLabel;
