// @mantine/core/cjs/components/LoadingOverlay/LoadingOverlay.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_default_z_index = require("../../core/utils/get-default-z-index/get-default-z-index.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_MantineThemeProvider = require("../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Overlay = require("../Overlay/Overlay.cjs");
const require_Transition = require("../Transition/Transition.cjs");
const require_Loader = require("../Loader/Loader.cjs");
const require_LoadingOverlay_module = require("./LoadingOverlay.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/LoadingOverlay/LoadingOverlay.tsx
const defaultProps = {
	transitionProps: {
		transition: "fade",
		duration: 0
	},
	overlayProps: { backgroundOpacity: .75 },
	zIndex: require_get_default_z_index.getDefaultZIndex("overlay")
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { zIndex }) => ({ root: { "--lo-z-index": zIndex?.toString() } }));
const LoadingOverlay = require_factory.factory((_props) => {
	const props = require_use_props.useProps("LoadingOverlay", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, transitionProps, loaderProps, overlayProps, visible, zIndex, attributes, onEnter, onEntered, onExit, onExited, ...others } = props;
	const theme = require_MantineThemeProvider.useMantineTheme();
	const getStyles = require_use_styles.useStyles({
		name: "LoadingOverlay",
		classes: require_LoadingOverlay_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const _overlayProps = {
		...defaultProps.overlayProps,
		...overlayProps
	};
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Transition.Transition, {
		transition: "fade",
		...transitionProps,
		mounted: !!visible,
		onEnter,
		onEntered,
		onExit,
		onExited,
		children: (transitionStyles) => /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
			...getStyles("root", { style: transitionStyles }),
			...others,
			children: [
				/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Loader.Loader, {
					unstyled,
					...loaderProps,
					...getStyles("loader", {
						className: loaderProps?.className,
						style: loaderProps?.style
					})
				}),
				/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Overlay.Overlay, {
					..._overlayProps,
					...getStyles("overlay", {
						className: _overlayProps?.className,
						style: _overlayProps?.style
					}),
					darkHidden: true,
					unstyled,
					color: overlayProps?.color || theme.white
				}),
				/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Overlay.Overlay, {
					..._overlayProps,
					...getStyles("overlay", {
						className: _overlayProps?.className,
						style: _overlayProps?.style
					}),
					lightHidden: true,
					unstyled,
					color: overlayProps?.color || theme.colors.dark[5]
				})
			]
		})
	});
});
LoadingOverlay.classes = require_LoadingOverlay_module.default;
LoadingOverlay.varsResolver = varsResolver;
LoadingOverlay.displayName = "@mantine/core/LoadingOverlay";
//#endregion
exports.LoadingOverlay = LoadingOverlay;
