// @mantine/core/cjs/components/Tabs/TabsList/TabsList.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Tabs_context = require("../Tabs.context.cjs");
const require_Tabs_module = require("../Tabs.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Tabs/TabsList/TabsList.tsx
const TabsList = require_factory.factory((_props) => {
	const props = require_use_props.useProps("TabsList", null, _props);
	const { children, className, grow, justify, classNames, styles, style, mod, ...others } = props;
	const ctx = require_Tabs_context.useTabsContext();
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...ctx.getStyles("list", {
			className,
			style,
			classNames,
			styles,
			props,
			variant: ctx.variant
		}),
		role: "tablist",
		variant: ctx.variant,
		mod: [{
			grow,
			orientation: ctx.orientation,
			placement: ctx.orientation === "vertical" && ctx.placement,
			inverted: ctx.inverted
		}, mod],
		"aria-orientation": ctx.orientation,
		__vars: { "--tabs-justify": justify },
		...others,
		children
	});
});
TabsList.classes = require_Tabs_module.default;
TabsList.displayName = "@mantine/core/TabsList";
//#endregion
exports.TabsList = TabsList;
