// @mantine/core/cjs/components/Flex/Flex.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_filter_props = require("../../core/utils/filter-props/filter-props.cjs");
const require_Mantine_context = require("../../core/MantineProvider/Mantine.context.cjs");
const require_MantineThemeProvider = require("../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_InlineStyles = require("../../core/InlineStyles/InlineStyles.cjs");
const require_hash_styles = require("../../core/InlineStyles/hash-styles.cjs");
const require_parse_style_props = require("../../core/Box/style-props/parse-style-props/parse-style-props.cjs");
const require_use_random_classname = require("../../core/Box/use-random-classname/use-random-classname.cjs");
const require_polymorphic_factory = require("../../core/factory/polymorphic-factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_flex_props = require("./flex-props.cjs");
const require_Flex_module = require("./Flex.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Flex/Flex.tsx
const Flex = require_polymorphic_factory.polymorphicFactory((_props) => {
	const props = require_use_props.useProps("Flex", null, _props);
	const { classNames, className, style, styles, unstyled, vars, gap, rowGap, columnGap, align, justify, wrap, direction, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Flex",
		classes: require_Flex_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars
	});
	const theme = require_MantineThemeProvider.useMantineTheme();
	const randomClassName = require_use_random_classname.useRandomClassName();
	const parsedStyleProps = require_parse_style_props.parseStyleProps({
		styleProps: {
			gap,
			rowGap,
			columnGap,
			align,
			justify,
			wrap,
			direction
		},
		theme,
		data: require_flex_props.FLEX_STYLE_PROPS_DATA
	});
	const deduplicateInlineStyles = require_Mantine_context.useMantineDeduplicateInlineStyles();
	const responsiveClassName = deduplicateInlineStyles && parsedStyleProps.hasResponsiveStyles ? require_hash_styles.hashStyleProps(parsedStyleProps.styles, parsedStyleProps.media) : randomClassName;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(react_jsx_runtime.Fragment, { children: [parsedStyleProps.hasResponsiveStyles && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_InlineStyles.InlineStyles, {
		selector: `.${responsiveClassName}`,
		styles: parsedStyleProps.styles,
		media: parsedStyleProps.media,
		deduplicate: deduplicateInlineStyles
	}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...getStyles("root", {
			className: responsiveClassName,
			style: require_filter_props.filterProps(parsedStyleProps.inlineStyles)
		}),
		...others
	})] });
});
Flex.classes = require_Flex_module.default;
Flex.displayName = "@mantine/core/Flex";
//#endregion
exports.Flex = Flex;
