// @mantine/core/cjs/utils/InlineInput/InlineInput.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Input = require("../../components/Input/Input.cjs");
const require_InlineInput_module = require("./InlineInput.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/utils/InlineInput/InlineInput.tsx
const InlineInputClasses = require_InlineInput_module.default;
function InlineInput({ __staticSelector, __stylesApiProps, className, classNames, styles, unstyled, children, label, description, id, disabled, error, size, labelPosition = "left", bodyElement = "div", labelElement = "label", variant, style, vars, mod, attributes, ...others }) {
	const getStyles = require_use_styles.useStyles({
		name: __staticSelector,
		props: __stylesApiProps,
		className,
		style,
		classes: require_InlineInput_module.default,
		classNames,
		styles,
		unstyled,
		attributes
	});
	const descriptionId = description ? `${id}-description` : void 0;
	const errorId = error && typeof error !== "boolean" ? `${id}-error` : void 0;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...getStyles("root"),
		__vars: {
			"--label-fz": require_get_size.getFontSize(size),
			"--label-lh": require_get_size.getSize(size, "label-lh")
		},
		mod: [{ "label-position": labelPosition }, mod],
		variant,
		size,
		...others,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
			component: bodyElement,
			htmlFor: bodyElement === "label" ? id : void 0,
			...getStyles("body"),
			children: [children, /* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
				...getStyles("labelWrapper"),
				"data-disabled": disabled || void 0,
				children: [
					label && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
						component: labelElement,
						htmlFor: labelElement === "label" ? id : void 0,
						...getStyles("label"),
						"data-disabled": disabled || void 0,
						children: label
					}),
					description && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Input.Input.Description, {
						id: descriptionId,
						size,
						__inheritStyles: false,
						...getStyles("description"),
						children: description
					}),
					error && typeof error !== "boolean" && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Input.Input.Error, {
						id: errorId,
						size,
						__inheritStyles: false,
						...getStyles("error"),
						children: error
					})
				]
			})]
		})
	});
}
InlineInput.displayName = "@mantine/core/InlineInput";
//#endregion
exports.InlineInput = InlineInput;
exports.InlineInputClasses = InlineInputClasses;
