// @mantine/core/cjs/components/ColorInput/ColorInput.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_resolved_styles_api = require("../../core/styles-api/use-resolved-styles-api/use-resolved-styles-api.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Popover = require("../Popover/Popover.cjs");
const require_ActionIcon = require("../ActionIcon/ActionIcon.cjs");
const require_Input = require("../Input/Input.cjs");
const require_use_input_props = require("../Input/use-input-props.cjs");
const require_InputBase = require("../InputBase/InputBase.cjs");
const require_ColorSwatch = require("../ColorSwatch/ColorSwatch.cjs");
const require_parsers = require("../ColorPicker/converters/parsers.cjs");
const require_converters = require("../ColorPicker/converters/converters.cjs");
const require_ColorPicker = require("../ColorPicker/ColorPicker.cjs");
const require_EyeDropperIcon = require("./EyeDropperIcon.cjs");
const require_ColorInput_module = require("./ColorInput.module.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/ColorInput/ColorInput.tsx
const defaultProps = {
	format: "hex",
	fixOnBlur: true,
	withPreview: true,
	swatchesPerRow: 7,
	withPicker: true,
	popoverProps: { transitionProps: {
		transition: "fade",
		duration: 0
	} },
	withEyeDropper: true,
	size: "sm",
	leftSectionPointerEvents: "none"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { size }) => ({
	eyeDropperIcon: { "--ci-eye-dropper-icon-size": require_get_size.getSize(size, "ci-eye-dropper-icon-size") },
	eyeDropperButton: { "--ci-button-size": require_get_size.getSize(size, "ci-button-size") },
	colorPreview: { "--ci-preview-size": require_get_size.getSize(size, "ci-preview-size") }
}));
const ColorInput = require_factory.factory((_props) => {
	const props = require_use_props.useProps([
		"Input",
		"InputWrapper",
		"ColorInput"
	], defaultProps, _props);
	const { classNames, styles, unstyled, disallowInput, fixOnBlur, popoverProps, withPreview, withEyeDropper, eyeDropperIcon, closeOnColorSwatchClick, eyeDropperButtonProps, value, defaultValue, onChange, onChangeEnd, onClick, onFocus, onBlur, inputProps, format = "hex", wrapperProps, readOnly, withPicker, swatches, disabled, leftSection, rightSection, swatchesPerRow, ...others } = require_use_input_props.useInputProps("ColorInput", defaultProps, _props);
	const getStyles = require_use_styles.useStyles({
		name: "ColorInput",
		props,
		classes: require_ColorInput_module.default,
		classNames,
		styles,
		unstyled,
		rootSelector: "wrapper",
		vars: props.vars,
		varsResolver
	});
	const { resolvedClassNames, resolvedStyles } = require_use_resolved_styles_api.useResolvedStylesApi({
		classNames,
		styles,
		props
	});
	const [dropdownOpened, setDropdownOpened] = (0, react.useState)(false);
	const [lastValidValue, setLastValidValue] = (0, react.useState)("");
	const [_value, setValue] = (0, _mantine_hooks.useUncontrolled)({
		value,
		defaultValue,
		finalValue: "",
		onChange
	});
	const { supported: eyeDropperSupported, open: openEyeDropper } = (0, _mantine_hooks.useEyeDropper)();
	const eyeDropper = /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_ActionIcon.ActionIcon, {
		...eyeDropperButtonProps,
		...getStyles("eyeDropperButton", {
			className: eyeDropperButtonProps?.className,
			style: eyeDropperButtonProps?.style
		}),
		variant: "subtle",
		color: "gray",
		unstyled,
		onClick: () => openEyeDropper().then((payload) => {
			if (payload?.sRGBHex) {
				const color = require_converters.convertHsvaTo(format, require_parsers.parseColor(payload.sRGBHex));
				setValue(color);
				onChangeEnd?.(color);
			}
		}).catch(() => {}),
		children: eyeDropperIcon || /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_EyeDropperIcon.EyeDropperIcon, { ...getStyles("eyeDropperIcon") })
	});
	const handleInputFocus = (event) => {
		onFocus?.(event);
		setDropdownOpened(true);
	};
	const handleInputBlur = (event) => {
		fixOnBlur && setValue(lastValidValue);
		onBlur?.(event);
		setDropdownOpened(false);
	};
	const handleInputClick = (event) => {
		onClick?.(event);
		setDropdownOpened(true);
	};
	(0, react.useEffect)(() => {
		if (require_parsers.isColorValid(_value) || _value.trim() === "") setLastValidValue(_value);
	}, [_value]);
	(0, _mantine_hooks.useDidUpdate)(() => {
		if (require_parsers.isColorValid(_value)) setValue(require_converters.convertHsvaTo(format, require_parsers.parseColor(_value)));
	}, [format]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Input.Input.Wrapper, {
		...wrapperProps,
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		__staticSelector: "ColorInput",
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Popover.Popover, {
			__staticSelector: "ColorInput",
			position: "bottom-start",
			offset: 5,
			opened: dropdownOpened,
			...popoverProps,
			classNames: resolvedClassNames,
			styles: resolvedStyles,
			unstyled,
			withRoles: false,
			disabled: readOnly || withPicker === false && (!Array.isArray(swatches) || swatches.length === 0),
			children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Popover.Popover.Target, { children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Input.Input, {
				autoComplete: "off",
				...others,
				...inputProps,
				classNames: resolvedClassNames,
				styles: resolvedStyles,
				disabled,
				__staticSelector: "ColorInput",
				onFocus: handleInputFocus,
				onBlur: handleInputBlur,
				onClick: handleInputClick,
				spellCheck: false,
				value: _value,
				onChange: (event) => {
					const inputValue = event.currentTarget.value;
					setValue(inputValue);
					if (require_parsers.isColorValid(inputValue)) onChangeEnd?.(require_converters.convertHsvaTo(format, require_parsers.parseColor(inputValue)));
				},
				leftSection: leftSection || (withPreview ? /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_ColorSwatch.ColorSwatch, {
					color: require_parsers.isColorValid(_value) ? _value : "#fff",
					size: "var(--ci-preview-size)",
					...getStyles("colorPreview")
				}) : null),
				readOnly: disallowInput || readOnly,
				pointer: disallowInput,
				unstyled,
				rightSection: rightSection || (withEyeDropper && !disabled && !readOnly && eyeDropperSupported ? eyeDropper : null)
			}) }), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Popover.Popover.Dropdown, {
				onMouseDown: (event) => event.preventDefault(),
				className: require_ColorInput_module.default.dropdown,
				children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_ColorPicker.ColorPicker, {
					__staticSelector: "ColorInput",
					value: _value,
					onChange: setValue,
					onChangeEnd,
					format,
					swatches,
					swatchesPerRow,
					withPicker,
					size: inputProps.size,
					focusable: false,
					unstyled,
					styles: resolvedStyles,
					classNames: resolvedClassNames,
					onColorSwatchClick: () => closeOnColorSwatchClick && setDropdownOpened(false),
					attributes: wrapperProps.attributes
				})
			})]
		})
	});
});
ColorInput.classes = require_InputBase.InputBase.classes;
ColorInput.varsResolver = varsResolver;
ColorInput.displayName = "@mantine/core/ColorInput";
//#endregion
exports.ColorInput = ColorInput;
