// @mantine/core/cjs/components/Chip/Chip.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_extract_style_props = require("../../core/Box/style-props/extract-style-props/extract-style-props.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_CheckIcon = require("../Checkbox/CheckIcon.cjs");
const require_ChipGroup = require("./ChipGroup/ChipGroup.cjs");
const require_Chip_module = require("./Chip.module.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Chip/Chip.tsx
const defaultProps = { type: "checkbox" };
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { size, radius, variant, color, autoContrast }) => {
	const colors = theme.variantColorResolver({
		color: color || theme.primaryColor,
		theme,
		variant: variant || "filled",
		autoContrast
	});
	return { root: {
		"--chip-fz": require_get_size.getFontSize(size),
		"--chip-size": require_get_size.getSize(size, "chip-size"),
		"--chip-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--chip-checked-padding": require_get_size.getSize(size, "chip-checked-padding"),
		"--chip-padding": require_get_size.getSize(size, "chip-padding"),
		"--chip-icon-size": require_get_size.getSize(size, "chip-icon-size"),
		"--chip-bg": color || variant ? colors.background : void 0,
		"--chip-hover": color || variant ? colors.hover : void 0,
		"--chip-color": color || variant ? colors.color : void 0,
		"--chip-bd": color || variant ? colors.border : void 0,
		"--chip-spacing": require_get_size.getSize(size, "chip-spacing")
	} };
});
const Chip = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Chip", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, id, checked, defaultChecked, onChange, value, wrapperProps, type, disabled, children, size, variant, icon, rootRef, autoContrast, mod, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Chip",
		classes: require_Chip_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const ctx = (0, react.use)(require_ChipGroup.ChipGroupContext);
	const uuid = (0, _mantine_hooks.useId)(id);
	const { styleProps, rest } = require_extract_style_props.extractStyleProps(others);
	const [_value, setValue] = (0, _mantine_hooks.useUncontrolled)({
		value: checked,
		defaultValue: defaultChecked,
		finalValue: false,
		onChange
	});
	const contextProps = ctx ? {
		checked: ctx.isChipSelected(value),
		onChange: (event) => {
			ctx.onChange(event);
			onChange?.(event.currentTarget.checked);
		},
		type: ctx.multiple ? "checkbox" : "radio"
	} : {};
	const _checked = contextProps.checked || _value;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		size,
		variant,
		ref: rootRef,
		mod,
		...getStyles("root"),
		...styleProps,
		...wrapperProps,
		children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)("input", {
			type,
			...getStyles("input"),
			checked: _checked,
			onChange: (event) => setValue(event.currentTarget.checked),
			id: uuid,
			disabled,
			value,
			...contextProps,
			...rest
		}), /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
			component: "label",
			htmlFor: uuid,
			mod: {
				checked: _checked,
				disabled
			},
			...getStyles("label", { variant: variant || "filled" }),
			children: [_checked && icon !== null && icon !== false && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
				...getStyles("iconWrapper"),
				children: icon === void 0 ? /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_CheckIcon.CheckIcon, { ...getStyles("checkIcon") }) : icon
			}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", { children })]
		})]
	});
});
Chip.classes = require_Chip_module.default;
Chip.varsResolver = varsResolver;
Chip.displayName = "@mantine/core/Chip";
Chip.Group = require_ChipGroup.ChipGroup;
//#endregion
exports.Chip = Chip;
