// @mantine/core/cjs/components/Avatar/Avatar.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_polymorphic_factory = require("../../core/factory/polymorphic-factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Avatar_module = require("./Avatar.module.cjs");
const require_AvatarGroup = require("./AvatarGroup/AvatarGroup.cjs");
const require_AvatarPlaceholderIcon = require("./AvatarPlaceholderIcon.cjs");
const require_get_initials_color = require("./get-initials-color/get-initials-color.cjs");
const require_get_initials = require("./get-initials/get-initials.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Avatar/Avatar.tsx
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { size, radius, variant, gradient, color, autoContrast, name, allowedInitialsColors }) => {
	const _color = color === "initials" && typeof name === "string" ? require_get_initials_color.getInitialsColor(name, allowedInitialsColors) : color;
	const colors = theme.variantColorResolver({
		color: _color || "gray",
		theme,
		gradient,
		variant: variant || "light",
		autoContrast
	});
	return { root: {
		"--avatar-size": require_get_size.getSize(size, "avatar-size"),
		"--avatar-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--avatar-bg": _color || variant ? colors.background : void 0,
		"--avatar-color": _color || variant ? colors.color : void 0,
		"--avatar-bd": _color || variant ? colors.border : void 0
	} };
});
const Avatar = require_polymorphic_factory.polymorphicFactory((_props) => {
	const props = require_use_props.useProps("Avatar", null, _props);
	const { classNames, className, style, styles, unstyled, vars, src, alt, radius, color, gradient, imageProps, children, autoContrast, mod, name, allowedInitialsColors, attributes, ...others } = props;
	const groupCtx = (0, react.use)(require_AvatarGroup.AvatarGroupContext);
	const [error, setError] = (0, react.useState)(!src);
	const getStyles = require_use_styles.useStyles({
		name: "Avatar",
		props,
		classes: require_Avatar_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	(0, react.useEffect)(() => setError(!src), [src]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...getStyles("root"),
		mod: [{ "within-group": groupCtx.withinGroup }, mod],
		...others,
		children: error || !src ? /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
			...getStyles("placeholder"),
			title: alt,
			children: children || typeof name === "string" && require_get_initials.getInitials(name) || /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_AvatarPlaceholderIcon.AvatarPlaceholderIcon, {})
		}) : /* @__PURE__ */ (0, react_jsx_runtime.jsx)("img", {
			...imageProps,
			...getStyles("image"),
			src,
			alt,
			onError: (event) => {
				setError(true);
				imageProps?.onError?.(event);
			}
		})
	});
});
Avatar.classes = require_Avatar_module.default;
Avatar.varsResolver = varsResolver;
Avatar.displayName = "@mantine/core/Avatar";
Avatar.Group = require_AvatarGroup.AvatarGroup;
//#endregion
exports.Avatar = Avatar;
