// @mantine/core/cjs/components/Accordion/AccordionPanel/AccordionPanel.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Collapse = require("../../Collapse/Collapse.cjs");
const require_Accordion_context = require("../Accordion.context.cjs");
const require_AccordionItem_context = require("../AccordionItem.context.cjs");
const require_Accordion_module = require("../Accordion.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Accordion/AccordionPanel/AccordionPanel.tsx
const AccordionPanel = require_factory.factory((props) => {
	const { classNames, className, style, styles, vars, children, keepMounted, ...others } = require_use_props.useProps("AccordionPanel", null, props);
	const { value } = require_AccordionItem_context.useAccordionItemContext();
	const ctx = require_Accordion_context.useAccordionContext();
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Collapse.Collapse, {
		...ctx.getStyles("panel", {
			className,
			classNames,
			style,
			styles
		}),
		expanded: ctx.isItemActive(value),
		transitionDuration: ctx.transitionDuration ?? 200,
		role: "region",
		id: ctx.getRegionId(value),
		"aria-labelledby": ctx.getControlId(value),
		keepMounted: keepMounted ?? ctx.keepMounted,
		...others,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
			...ctx.getStyles("content", {
				classNames,
				styles
			}),
			children
		})
	});
});
AccordionPanel.displayName = "@mantine/core/AccordionPanel";
AccordionPanel.classes = require_Accordion_module.default;
//#endregion
exports.AccordionPanel = AccordionPanel;
