// @mantine/core/cjs/components/List/List.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_List_context = require("./List.context.cjs");
const require_List_module = require("./List.module.cjs");
const require_ListItem = require("./ListItem/ListItem.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/List/List.tsx
const defaultProps = { type: "unordered" };
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { size, spacing }) => ({ root: {
	"--list-fz": require_get_size.getFontSize(size),
	"--list-lh": require_get_size.getLineHeight(size),
	"--list-spacing": require_get_size.getSpacing(spacing)
} }));
const List = require_factory.factory((_props) => {
	const props = require_use_props.useProps("List", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, children, type, withPadding, icon, spacing, center, listStyleType, mod, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "List",
		classes: require_List_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_List_context.ListProvider, {
		value: {
			center,
			icon,
			getStyles
		},
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			...getStyles("root", { style: { listStyleType } }),
			component: type === "unordered" ? "ul" : "ol",
			mod: [{
				"with-padding": withPadding,
				type: icon ? "none" : listStyleType
			}, mod],
			...others,
			children
		})
	});
});
List.classes = require_List_module.default;
List.varsResolver = varsResolver;
List.displayName = "@mantine/core/List";
List.Item = require_ListItem.ListItem;
//#endregion
exports.List = List;
