// @mantine/core/cjs/components/CloseButton/CloseButton.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../core/utils/units-converters/rem.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_polymorphic_factory = require("../../core/factory/polymorphic-factory.cjs");
const require_UnstyledButton = require("../UnstyledButton/UnstyledButton.cjs");
const require_CloseIcon = require("./CloseIcon.cjs");
const require_CloseButton_module = require("./CloseButton.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/CloseButton/CloseButton.tsx
const defaultProps = { variant: "subtle" };
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { size, radius, iconSize }) => ({ root: {
	"--cb-size": require_get_size.getSize(size, "cb-size"),
	"--cb-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
	"--cb-icon-size": require_rem.rem(iconSize)
} }));
const CloseButton = require_polymorphic_factory.polymorphicFactory((_props) => {
	const props = require_use_props.useProps("CloseButton", defaultProps, _props);
	const { iconSize, children, vars, radius, className, classNames, style, styles, unstyled, "data-disabled": dataDisabled, disabled, variant, icon, mod, attributes, __staticSelector, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: __staticSelector || "CloseButton",
		props,
		className,
		style,
		classes: require_CloseButton_module.default,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_UnstyledButton.UnstyledButton, {
		...others,
		unstyled,
		variant,
		disabled,
		mod: [{ disabled: disabled || dataDisabled }, mod],
		...getStyles("root", {
			variant,
			active: !disabled && !dataDisabled
		}),
		children: [icon || /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_CloseIcon.CloseIcon, {}), children]
	});
});
CloseButton.classes = require_CloseButton_module.default;
CloseButton.varsResolver = varsResolver;
CloseButton.displayName = "@mantine/core/CloseButton";
//#endregion
exports.CloseButton = CloseButton;
