// @mantine/core/cjs/core/InlineStyles/InlineStyles.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_Mantine_context = require("../MantineProvider/Mantine.context.cjs");
const require_styles_to_string = require("./styles-to-string/styles-to-string.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/core/InlineStyles/InlineStyles.tsx
function simpleHash(str) {
	let hash = 5381;
	for (let i = 0; i < str.length; i++) hash = (hash << 5) + hash + str.charCodeAt(i) & 4294967295;
	return (hash >>> 0).toString(36);
}
function InlineStyles({ deduplicate, ...props }) {
	const nonce = require_Mantine_context.useMantineStyleNonce();
	const css = require_styles_to_string.stylesToString(props);
	if (deduplicate) return /* @__PURE__ */ (0, react_jsx_runtime.jsx)("style", {
		href: `mantine-${simpleHash(css)}`,
		precedence: "mantine",
		nonce: nonce?.(),
		children: css
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)("style", {
		"data-mantine-styles": "inline",
		nonce: nonce?.(),
		dangerouslySetInnerHTML: { __html: css }
	});
}
//#endregion
exports.InlineStyles = InlineStyles;
