// @mantine/core/cjs/components/Pagination/PaginationRoot/PaginationRoot.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_event_handler = require("../../../core/utils/create-event-handler/create-event-handler.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_get_contrast_color = require("../../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs");
const require_get_auto_contrast_value = require("../../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Pagination_context = require("../Pagination.context.cjs");
const require_Pagination_module = require("../Pagination.module.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Pagination/PaginationRoot/PaginationRoot.tsx
const defaultProps = {
	siblings: 1,
	boundaries: 1
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { size, radius, color, autoContrast }) => ({ root: {
	"--pagination-control-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
	"--pagination-control-size": require_get_size.getSize(size, "pagination-control-size"),
	"--pagination-control-fz": require_get_size.getFontSize(size),
	"--pagination-active-bg": color ? require_get_theme_color.getThemeColor(color, theme) : void 0,
	"--pagination-active-color": require_get_auto_contrast_value.getAutoContrastValue(autoContrast, theme) ? require_get_contrast_color.getContrastColor({
		color,
		theme,
		autoContrast
	}) : void 0
} }));
const PaginationRoot = require_factory.factory((_props) => {
	const props = require_use_props.useProps("PaginationRoot", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, total, value, defaultValue, onChange, disabled, siblings, boundaries, color, radius, onNextPage, onPreviousPage, onFirstPage, onLastPage, getItemProps, autoContrast, startValue, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Pagination",
		classes: require_Pagination_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const { range, setPage, next, previous, active, first, last } = (0, _mantine_hooks.usePagination)({
		page: value,
		initialPage: defaultValue,
		onChange,
		total,
		siblings,
		boundaries,
		startValue
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Pagination_context.PaginationProvider, {
		value: {
			total,
			range,
			active,
			disabled,
			getItemProps,
			onChange: setPage,
			onNext: require_create_event_handler.createEventHandler(onNextPage, next),
			onPrevious: require_create_event_handler.createEventHandler(onPreviousPage, previous),
			onFirst: require_create_event_handler.createEventHandler(onFirstPage, first),
			onLast: require_create_event_handler.createEventHandler(onLastPage, last),
			getStyles
		},
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			...getStyles("root"),
			...others
		})
	});
});
PaginationRoot.classes = require_Pagination_module.default;
PaginationRoot.varsResolver = varsResolver;
PaginationRoot.displayName = "@mantine/core/PaginationRoot";
//#endregion
exports.PaginationRoot = PaginationRoot;
