// @mantine/core/cjs/components/RingProgress/Curve/Curve.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_theme_color = require("../../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_MantineThemeProvider = require("../../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Tooltip = require("../../Tooltip/Tooltip.cjs");
const require_get_curve_props = require("./get-curve-props.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/RingProgress/Curve/Curve.tsx
function Curve({ size, value, offset, sum, thickness, root, color, lineRoundCaps, tooltip, getStyles, display, ...others }) {
	const theme = require_MantineThemeProvider.useMantineTheme();
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Tooltip.Tooltip.Floating, {
		disabled: !tooltip,
		label: tooltip,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			component: "circle",
			...others,
			...getStyles("curve"),
			__vars: { "--curve-color": color ? require_get_theme_color.getThemeColor(color, theme) : void 0 },
			fill: "none",
			strokeLinecap: lineRoundCaps ? "round" : "butt",
			...require_get_curve_props.getCurveProps({
				sum,
				size,
				thickness,
				value,
				offset,
				root
			})
		})
	});
}
Curve.displayName = "@mantine/core/Curve";
//#endregion
exports.Curve = Curve;
