// @mantine/hooks/esm/index.mjs
import { clamp } from "./utils/clamp/clamp.mjs";
import { lowerFirst } from "./utils/lower-first/lower-first.mjs";
import { randomId } from "./utils/random-id/random-id.mjs";
import { range } from "./utils/range/range.mjs";
import { shallowEqual } from "./utils/shallow-equal/shallow-equal.mjs";
import { upperFirst } from "./utils/upper-first/upper-first.mjs";
import { useCallbackRef } from "./utils/use-callback-ref/use-callback-ref.mjs";
import { useDebouncedCallback } from "./use-debounced-callback/use-debounced-callback.mjs";
import { useClickOutside } from "./use-click-outside/use-click-outside.mjs";
import { useClipboard } from "./use-clipboard/use-clipboard.mjs";
import { useMediaQuery } from "./use-media-query/use-media-query.mjs";
import { useColorScheme } from "./use-color-scheme/use-color-scheme.mjs";
import { useCounter } from "./use-counter/use-counter.mjs";
import { useDebouncedState } from "./use-debounced-state/use-debounced-state.mjs";
import { useDebouncedValue } from "./use-debounced-value/use-debounced-value.mjs";
import { useIsomorphicEffect } from "./use-isomorphic-effect/use-isomorphic-effect.mjs";
import { useDocumentTitle } from "./use-document-title/use-document-title.mjs";
import { useDocumentVisibility } from "./use-document-visibility/use-document-visibility.mjs";
import { useDidUpdate } from "./use-did-update/use-did-update.mjs";
import { useFocusReturn } from "./use-focus-return/use-focus-return.mjs";
import { useFocusTrap } from "./use-focus-trap/use-focus-trap.mjs";
import { useForceUpdate } from "./use-force-update/use-force-update.mjs";
import { useId } from "./use-id/use-id.mjs";
import { useIdle } from "./use-idle/use-idle.mjs";
import { useInterval } from "./use-interval/use-interval.mjs";
import { useListState } from "./use-list-state/use-list-state.mjs";
import { useWindowEvent } from "./use-window-event/use-window-event.mjs";
import { readLocalStorageValue, useLocalStorage } from "./use-local-storage/use-local-storage.mjs";
import { readSessionStorageValue, useSessionStorage } from "./use-session-storage/use-session-storage.mjs";
import { assignRef, mergeRefs, useMergedRef } from "./use-merged-ref/use-merged-ref.mjs";
import { useMouse, useMousePosition } from "./use-mouse/use-mouse.mjs";
import { clampUseMovePosition, useMove } from "./use-move/use-move.mjs";
import { useUncontrolled } from "./use-uncontrolled/use-uncontrolled.mjs";
import { usePagination } from "./use-pagination/use-pagination.mjs";
import { useQueue } from "./use-queue/use-queue.mjs";
import { usePageLeave } from "./use-page-leave/use-page-leave.mjs";
import { useReducedMotion } from "./use-reduced-motion/use-reduced-motion.mjs";
import { useScrollIntoView } from "./use-scroll-into-view/use-scroll-into-view.mjs";
import { useElementSize, useResizeObserver } from "./use-resize-observer/use-resize-observer.mjs";
import { useShallowEffect } from "./use-shallow-effect/use-shallow-effect.mjs";
import { useToggle } from "./use-toggle/use-toggle.mjs";
import { useViewportSize } from "./use-viewport-size/use-viewport-size.mjs";
import { useWindowScroll } from "./use-window-scroll/use-window-scroll.mjs";
import { useIntersection } from "./use-intersection/use-intersection.mjs";
import { useHash } from "./use-hash/use-hash.mjs";
import { getHotkeyHandler } from "./use-hotkeys/parse-hotkey.mjs";
import { useHotkeys } from "./use-hotkeys/use-hotkeys.mjs";
import { useFullscreenDocument, useFullscreenElement } from "./use-fullscreen/use-fullscreen.mjs";
import { useLogger } from "./use-logger/use-logger.mjs";
import { useHover } from "./use-hover/use-hover.mjs";
import { useValidatedState } from "./use-validated-state/use-validated-state.mjs";
import { useOs } from "./use-os/use-os.mjs";
import { useSetState } from "./use-set-state/use-set-state.mjs";
import { useInputState } from "./use-input-state/use-input-state.mjs";
import { useEventListener } from "./use-event-listener/use-event-listener.mjs";
import { useDisclosure } from "./use-disclosure/use-disclosure.mjs";
import { useFocusWithin } from "./use-focus-within/use-focus-within.mjs";
import { useNetwork } from "./use-network/use-network.mjs";
import { useTimeout } from "./use-timeout/use-timeout.mjs";
import { useTextSelection } from "./use-text-selection/use-text-selection.mjs";
import { usePrevious } from "./use-previous/use-previous.mjs";
import { useFavicon } from "./use-favicon/use-favicon.mjs";
import { useScrollDirection } from "./use-scroll-direction/use-scroll-direction.mjs";
import { useHeadroom } from "./use-headroom/use-headroom.mjs";
import { useEyeDropper } from "./use-eye-dropper/use-eye-dropper.mjs";
import { useInViewport } from "./use-in-viewport/use-in-viewport.mjs";
import { useMutationObserver, useMutationObserverTarget } from "./use-mutation-observer/use-mutation-observer.mjs";
import { useMounted } from "./use-mounted/use-mounted.mjs";
import { useStateHistory } from "./use-state-history/use-state-history.mjs";
import { useMap } from "./use-map/use-map.mjs";
import { useSet } from "./use-set/use-set.mjs";
import { useThrottledCallback } from "./use-throttled-callback/use-throttled-callback.mjs";
import { useThrottledState } from "./use-throttled-state/use-throttled-state.mjs";
import { useThrottledValue } from "./use-throttled-value/use-throttled-value.mjs";
import { useIsFirstRender } from "./use-is-first-render/use-is-first-render.mjs";
import { useOrientation } from "./use-orientation/use-orientation.mjs";
import { useFetch } from "./use-fetch/use-fetch.mjs";
import { normalizeRadialValue, useRadialMove } from "./use-radial-move/use-radial-move.mjs";
import { useScrollSpy } from "./use-scroll-spy/use-scroll-spy.mjs";
import { useScroller } from "./use-scroller/use-scroller.mjs";
import { useFileDialog } from "./use-file-dialog/use-file-dialog.mjs";
import { useLongPress } from "./use-long-press/use-long-press.mjs";
import { useSelection } from "./use-selection/use-selection.mjs";
import { useFloatingWindow } from "./use-floating-window/use-floating-window.mjs";
import { useCollapse } from "./use-collapse/use-collapse.mjs";
import { useHorizontalCollapse } from "./use-collapse/use-horizontal-collapse.mjs";
import { formatMask, generatePattern, isMaskComplete, unformatMask, useMask } from "./use-mask/use-mask.mjs";
import { useRovingIndex } from "./use-roving-index/use-roving-index.mjs";
import { useDrag } from "./use-drag/use-drag.mjs";
export { assignRef, clamp, clampUseMovePosition, formatMask, generatePattern, getHotkeyHandler, isMaskComplete, lowerFirst, mergeRefs, normalizeRadialValue, randomId, range, readLocalStorageValue, readSessionStorageValue, shallowEqual, unformatMask, upperFirst, useCallbackRef, useClickOutside, useClipboard, useCollapse, useColorScheme, useCounter, useDebouncedCallback, useDebouncedState, useDebouncedValue, useDidUpdate, useDisclosure, useDocumentTitle, useDocumentVisibility, useDrag, useElementSize, useEventListener, useEyeDropper, useFavicon, useFetch, useFileDialog, useFloatingWindow, useFocusReturn, useFocusTrap, useFocusWithin, useForceUpdate, useFullscreenDocument, useFullscreenElement, useHash, useHeadroom, useHorizontalCollapse, useHotkeys, useHover, useId, useIdle, useInViewport, useInputState, useIntersection, useInterval, useIsFirstRender, useIsomorphicEffect, useListState, useLocalStorage, useLogger, useLongPress, useMap, useMask, useMediaQuery, useMergedRef, useMounted, useMouse, useMousePosition, useMove, useMutationObserver, useMutationObserverTarget, useNetwork, useOrientation, useOs, usePageLeave, usePagination, usePrevious, useQueue, useRadialMove, useReducedMotion, useResizeObserver, useRovingIndex, useScrollDirection, useScrollIntoView, useScrollSpy, useScroller, useSelection, useSessionStorage, useSet, useSetState, useShallowEffect, useStateHistory, useTextSelection, useThrottledCallback, useThrottledState, useThrottledValue, useTimeout, useToggle, useUncontrolled, useValidatedState, useViewportSize, useWindowEvent, useWindowScroll };
