// @mantine/core/cjs/components/Spoiler/Spoiler.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../core/utils/units-converters/rem.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Anchor = require("../Anchor/Anchor.cjs");
const require_Spoiler_module = require("./Spoiler.module.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Spoiler/Spoiler.tsx
const defaultProps = {
	maxHeight: 100,
	defaultExpanded: false
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { transitionDuration }) => ({ root: { "--spoiler-transition-duration": transitionDuration !== void 0 ? `${transitionDuration}ms` : void 0 } }));
const Spoiler = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Spoiler", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, defaultExpanded, maxHeight, hideLabel, showLabel, children, controlRef, transitionDuration, id, expanded, onExpandedChange, showAriaLabel, hideAriaLabel, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Spoiler",
		classes: require_Spoiler_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const _id = (0, _mantine_hooks.useId)(id);
	const regionId = `${_id}-region`;
	const [show, setShowState] = (0, _mantine_hooks.useUncontrolled)({
		value: expanded,
		defaultValue: defaultExpanded,
		finalValue: false,
		onChange: onExpandedChange
	});
	const { ref: contentRef, height } = (0, _mantine_hooks.useElementSize)();
	const spoilerMoreContent = show ? hideLabel : showLabel;
	const spoiler = spoilerMoreContent !== null && maxHeight < height;
	const ariaLabel = show ? hideAriaLabel : showAriaLabel;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		...getStyles("root"),
		id: _id,
		"data-has-spoiler": spoiler || void 0,
		...others,
		children: [spoiler && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Anchor.Anchor, {
			component: "button",
			type: "button",
			ref: controlRef,
			onClick: () => setShowState(!show),
			"aria-expanded": show,
			"aria-controls": regionId,
			"aria-label": ariaLabel,
			...getStyles("control"),
			children: spoilerMoreContent
		}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
			...getStyles("content", { style: { maxHeight: !show ? require_rem.rem(maxHeight) : height ? require_rem.rem(height) : void 0 } }),
			"data-reduce-motion": true,
			role: "region",
			id: regionId,
			children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
				ref: contentRef,
				children
			})
		})]
	});
});
Spoiler.classes = require_Spoiler_module.default;
Spoiler.varsResolver = varsResolver;
Spoiler.displayName = "@mantine/core/Spoiler";
//#endregion
exports.Spoiler = Spoiler;
