// @mantine/core/cjs/components/Stack/Stack.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Stack_module = require("./Stack.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Stack/Stack.tsx
const defaultProps = {
	gap: "md",
	align: "stretch",
	justify: "flex-start"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { gap, align, justify }) => ({ root: {
	"--stack-gap": require_get_size.getSpacing(gap),
	"--stack-align": align,
	"--stack-justify": justify
} }));
const Stack = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Stack", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, align, justify, gap, variant, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...require_use_styles.useStyles({
			name: "Stack",
			props,
			classes: require_Stack_module.default,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver
		})("root"),
		variant,
		...others
	});
});
Stack.classes = require_Stack_module.default;
Stack.varsResolver = varsResolver;
Stack.displayName = "@mantine/core/Stack";
//#endregion
exports.Stack = Stack;
