// @mantine/core/cjs/components/TagsInput/TagsInput.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_resolved_styles_api = require("../../core/styles-api/use-resolved-styles-api/use-resolved-styles-api.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_extract_style_props = require("../../core/Box/style-props/extract-style-props/extract-style-props.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_InputBase = require("../InputBase/InputBase.cjs");
const require_get_parsed_combobox_data = require("../Combobox/get-parsed-combobox-data/get-parsed-combobox-data.cjs");
const require_get_options_lockup = require("../Combobox/get-options-lockup/get-options-lockup.cjs");
const require_use_combobox = require("../Combobox/use-combobox/use-combobox.cjs");
const require_Combobox = require("../Combobox/Combobox.cjs");
const require_OptionsDropdown = require("../Combobox/OptionsDropdown/OptionsDropdown.cjs");
const require_use_pills_reorder = require("../Combobox/use-pills-reorder/use-pills-reorder.cjs");
const require_Pill = require("../Pill/Pill.cjs");
const require_PillsInput = require("../PillsInput/PillsInput.cjs");
const require_filter_picked_tags = require("./filter-picked-tags.cjs");
const require_get_splitted_tags = require("./get-splitted-tags.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/TagsInput/TagsInput.tsx
const defaultProps = {
	maxTags: Infinity,
	acceptValueOnBlur: true,
	splitChars: [","],
	hiddenInputValuesDivider: ",",
	openOnFocus: true,
	size: "sm"
};
const TagsInput = require_factory.factory((_props) => {
	const props = require_use_props.useProps([
		"Input",
		"InputWrapper",
		"TagsInput"
	], defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, size, value, defaultValue, onChange, onKeyDown, maxTags, allowDuplicates, onDuplicate, variant, data, dropdownOpened, defaultDropdownOpened, onDropdownOpen, onDropdownClose, selectFirstOptionOnChange, selectFirstOptionOnDropdownOpen, onOptionSubmit, comboboxProps, filter, limit, withScrollArea, maxDropdownHeight, searchValue, defaultSearchValue, onSearchChange, readOnly, disabled, splitChars, onFocus, onBlur, onPaste, radius, rightSection, rightSectionWidth, rightSectionPointerEvents, rightSectionProps, leftSection, leftSectionWidth, leftSectionPointerEvents, leftSectionProps, inputContainer, inputWrapperOrder, withAsterisk, required, labelProps, descriptionProps, errorProps, wrapperProps, description, label, error, withErrorStyles, name, form, id, clearable, clearSectionMode, clearButtonProps, hiddenInputProps, hiddenInputValuesDivider, mod, renderOption, renderPill, onRemove, onClear, onMaxTags, scrollAreaProps, acceptValueOnBlur, isDuplicate, openOnFocus, attributes, ref, loading, loadingPosition, withPillsReorder, ...others } = props;
	const _id = (0, _mantine_hooks.useId)(id);
	const parsedData = require_get_parsed_combobox_data.getParsedComboboxData(data);
	const optionsLockup = require_get_options_lockup.getOptionsLockup(parsedData);
	const inputRef = (0, react.useRef)(null);
	const _ref = (0, _mantine_hooks.useMergedRef)(inputRef, ref);
	const combobox = require_use_combobox.useCombobox({
		opened: dropdownOpened,
		defaultOpened: defaultDropdownOpened,
		onDropdownOpen: () => {
			onDropdownOpen?.();
			if (selectFirstOptionOnDropdownOpen) combobox.selectFirstOption();
		},
		onDropdownClose: () => {
			onDropdownClose?.();
			combobox.resetSelectedOption();
		}
	});
	const { styleProps, rest: { type, autoComplete, ...rest } } = require_extract_style_props.extractStyleProps(others);
	const [_value, setValue] = (0, _mantine_hooks.useUncontrolled)({
		value,
		defaultValue,
		finalValue: [],
		onChange
	});
	const { getPillProps, getListProps, handleInputKeyDown } = require_use_pills_reorder.usePillsReorder({
		value: _value,
		onChange: setValue,
		enabled: withPillsReorder && !disabled && !readOnly
	});
	const [_searchValue, setSearchValue] = (0, _mantine_hooks.useUncontrolled)({
		value: searchValue,
		defaultValue: defaultSearchValue,
		finalValue: "",
		onChange: onSearchChange
	});
	const handleSearchChange = (value) => {
		setSearchValue(value);
		combobox.resetSelectedOption();
	};
	const getStyles = require_use_styles.useStyles({
		name: "TagsInput",
		classes: {},
		props,
		classNames,
		styles,
		unstyled
	});
	const { resolvedClassNames, resolvedStyles } = require_use_resolved_styles_api.useResolvedStylesApi({
		props,
		styles,
		classNames
	});
	const handleValueSelect = (val) => {
		if (isDuplicate ? isDuplicate(val, _value) : _value.some((tag) => tag.toLowerCase() === val.toLowerCase())) {
			onDuplicate?.(val);
			if (!allowDuplicates) {
				handleSearchChange("");
				return;
			}
		}
		if (_value.length >= maxTags) {
			onMaxTags?.(val);
			return;
		}
		onOptionSubmit?.(val);
		handleSearchChange("");
		if (val.length > 0) setValue([..._value, val]);
	};
	const handleInputKeydown = (event) => {
		onKeyDown?.(event);
		if (event.defaultPrevented || event.isPropagationStopped()) return;
		const inputValue = _searchValue.trim();
		const { length } = inputValue;
		if (splitChars.includes(event.key) && length > 0) {
			setValue(require_get_splitted_tags.getSplittedTags({
				splitChars,
				allowDuplicates,
				maxTags,
				value: _searchValue,
				currentTags: _value,
				isDuplicate,
				onDuplicate
			}));
			handleSearchChange("");
			event.preventDefault();
		}
		if (event.key === "Enter" && length > 0 && !event.nativeEvent.isComposing) {
			event.preventDefault();
			if (!!document.querySelector(`#${combobox.listId} [data-combobox-option][data-combobox-selected]`)) return;
			handleValueSelect(inputValue);
		}
		if (event.key === "Backspace" && length === 0 && _value.length > 0 && !event.nativeEvent.isComposing && !readOnly) {
			onRemove?.(_value[_value.length - 1]);
			setValue(_value.slice(0, _value.length - 1));
		}
		handleInputKeyDown(event);
	};
	const handlePaste = (event) => {
		onPaste?.(event);
		event.preventDefault();
		if (event.clipboardData) {
			setValue(require_get_splitted_tags.getSplittedTags({
				splitChars,
				allowDuplicates,
				maxTags,
				value: `${_searchValue}${event.clipboardData.getData("text/plain")}`,
				currentTags: _value,
				isDuplicate,
				onDuplicate
			}));
			handleSearchChange("");
		}
	};
	const values = _value.map((item, index) => {
		const onRemoveItem = () => {
			const next_value = _value.slice();
			next_value.splice(index, 1);
			setValue(next_value);
			onRemove?.(item);
		};
		const reorderProps = getPillProps(index);
		if (renderPill) return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(react.Fragment, { children: renderPill({
			option: optionsLockup[item] || {
				value: item,
				label: item,
				disabled: false
			},
			value: item,
			onRemove: onRemoveItem,
			disabled: disabled || readOnly,
			reorderProps
		}) }, `${item}-${index}`);
		return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Pill.Pill, {
			withRemoveButton: !readOnly,
			onRemove: onRemoveItem,
			unstyled,
			disabled,
			attributes,
			...getStyles("pill"),
			...reorderProps,
			children: item
		}, `${item}-${index}`);
	});
	(0, react.useEffect)(() => {
		if (selectFirstOptionOnChange) combobox.selectFirstOption();
	}, [
		selectFirstOptionOnChange,
		_value,
		_searchValue
	]);
	const clearButton = /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Combobox.Combobox.ClearButton, {
		...clearButtonProps,
		onClear: () => {
			setValue([]);
			handleSearchChange("");
			inputRef.current?.focus();
			combobox.openDropdown();
			onClear?.();
		}
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(react_jsx_runtime.Fragment, { children: [/* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Combobox.Combobox, {
		store: combobox,
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		unstyled,
		size,
		readOnly,
		__staticSelector: "TagsInput",
		attributes,
		onOptionSubmit: (val) => {
			onOptionSubmit?.(val);
			handleSearchChange("");
			if (_value.length >= maxTags) onMaxTags?.(val);
			else setValue([..._value, optionsLockup[val].value]);
			combobox.resetSelectedOption();
		},
		...comboboxProps,
		children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Combobox.Combobox.DropdownTarget, { children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_PillsInput.PillsInput, {
			...styleProps,
			__staticSelector: "TagsInput",
			classNames: resolvedClassNames,
			styles: resolvedStyles,
			unstyled,
			size,
			className,
			style,
			variant,
			disabled,
			radius,
			rightSection,
			__clearSection: clearButton,
			__clearable: clearable && _value.length > 0 && !disabled && !readOnly,
			__clearSectionMode: clearSectionMode,
			rightSectionWidth,
			rightSectionPointerEvents,
			rightSectionProps,
			leftSection,
			leftSectionWidth,
			leftSectionPointerEvents,
			leftSectionProps,
			loading,
			loadingPosition,
			inputContainer,
			inputWrapperOrder,
			withAsterisk,
			required,
			labelProps,
			descriptionProps,
			errorProps,
			wrapperProps,
			description,
			label,
			error,
			withErrorStyles,
			__stylesApiProps: {
				...props,
				multiline: true
			},
			id: _id,
			mod,
			attributes,
			children: /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Pill.Pill.Group, {
				disabled,
				unstyled,
				...getStyles("pillsList"),
				...getListProps(),
				children: [values, /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Combobox.Combobox.EventsTarget, {
					autoComplete,
					withExpandedAttribute: true,
					children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_PillsInput.PillsInput.Field, {
						...rest,
						ref: _ref,
						...getStyles("inputField"),
						unstyled,
						onKeyDown: handleInputKeydown,
						onFocus: (event) => {
							onFocus?.(event);
							openOnFocus && combobox.openDropdown();
						},
						onBlur: (event) => {
							onBlur?.(event);
							acceptValueOnBlur && handleValueSelect(_searchValue);
							combobox.closeDropdown();
						},
						onPaste: handlePaste,
						value: _searchValue,
						onChange: (event) => handleSearchChange(event.currentTarget.value),
						required: required && _value.length === 0,
						disabled,
						readOnly,
						id: _id
					})
				})]
			})
		}) }), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_OptionsDropdown.OptionsDropdown, {
			data: require_filter_picked_tags.filterPickedTags({
				data: parsedData,
				value: _value
			}),
			hidden: readOnly || disabled,
			filter,
			search: _searchValue,
			limit,
			hiddenWhenEmpty: true,
			withScrollArea,
			maxDropdownHeight,
			unstyled,
			labelId: label ? `${_id}-label` : void 0,
			"aria-label": label ? void 0 : others["aria-label"],
			renderOption,
			scrollAreaProps
		})]
	}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Combobox.Combobox.HiddenInput, {
		name,
		form,
		value: _value,
		valuesDivider: hiddenInputValuesDivider,
		disabled,
		...hiddenInputProps
	})] });
});
TagsInput.classes = {
	...require_InputBase.InputBase.classes,
	...require_Combobox.Combobox.classes
};
TagsInput.displayName = "@mantine/core/TagsInput";
//#endregion
exports.TagsInput = TagsInput;
