// @mantine/core/cjs/core/MantineProvider/default-theme.cjs
const require_rem = require("../utils/units-converters/rem.cjs");
const require_default_variant_colors_resolver = require("./color-functions/default-variant-colors-resolver/default-variant-colors-resolver.cjs");
const require_default_colors = require("./default-colors.cjs");
//#region packages/@mantine/core/src/core/MantineProvider/default-theme.ts
const DEFAULT_FONT_FAMILY = "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji";
const DEFAULT_THEME = {
	scale: 1,
	fontSmoothing: true,
	focusRing: "auto",
	white: "#fff",
	black: "#000",
	colors: require_default_colors.DEFAULT_COLORS,
	primaryShade: {
		light: 6,
		dark: 8
	},
	primaryColor: "blue",
	variantColorResolver: require_default_variant_colors_resolver.defaultVariantColorsResolver,
	autoContrast: false,
	luminanceThreshold: .3,
	fontFamily: DEFAULT_FONT_FAMILY,
	fontFamilyMonospace: "ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, Liberation Mono, Courier New, monospace",
	respectReducedMotion: false,
	cursorType: "default",
	defaultGradient: {
		from: "blue",
		to: "cyan",
		deg: 45
	},
	defaultRadius: "md",
	activeClassName: "mantine-active",
	focusClassName: "",
	headings: {
		fontFamily: DEFAULT_FONT_FAMILY,
		fontWeight: "700",
		textWrap: "wrap",
		sizes: {
			h1: {
				fontSize: require_rem.rem(34),
				lineHeight: "1.3"
			},
			h2: {
				fontSize: require_rem.rem(26),
				lineHeight: "1.35"
			},
			h3: {
				fontSize: require_rem.rem(22),
				lineHeight: "1.4"
			},
			h4: {
				fontSize: require_rem.rem(18),
				lineHeight: "1.45"
			},
			h5: {
				fontSize: require_rem.rem(16),
				lineHeight: "1.5"
			},
			h6: {
				fontSize: require_rem.rem(14),
				lineHeight: "1.5"
			}
		}
	},
	fontSizes: {
		xs: require_rem.rem(12),
		sm: require_rem.rem(14),
		md: require_rem.rem(16),
		lg: require_rem.rem(18),
		xl: require_rem.rem(20)
	},
	lineHeights: {
		xs: "1.4",
		sm: "1.45",
		md: "1.55",
		lg: "1.6",
		xl: "1.65"
	},
	fontWeights: {
		regular: "400",
		medium: "600",
		bold: "700"
	},
	radius: {
		xs: require_rem.rem(2),
		sm: require_rem.rem(4),
		md: require_rem.rem(8),
		lg: require_rem.rem(16),
		xl: require_rem.rem(32)
	},
	spacing: {
		xs: require_rem.rem(10),
		sm: require_rem.rem(12),
		md: require_rem.rem(16),
		lg: require_rem.rem(20),
		xl: require_rem.rem(32)
	},
	breakpoints: {
		xs: "36em",
		sm: "48em",
		md: "62em",
		lg: "75em",
		xl: "88em"
	},
	shadows: {
		xs: `0 ${require_rem.rem(1)} ${require_rem.rem(3)} rgba(0, 0, 0, 0.05), 0 ${require_rem.rem(1)} ${require_rem.rem(2)} rgba(0, 0, 0, 0.1)`,
		sm: `0 ${require_rem.rem(1)} ${require_rem.rem(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${require_rem.rem(10)} ${require_rem.rem(15)} ${require_rem.rem(-5)}, rgba(0, 0, 0, 0.04) 0 ${require_rem.rem(7)} ${require_rem.rem(7)} ${require_rem.rem(-5)}`,
		md: `0 ${require_rem.rem(1)} ${require_rem.rem(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${require_rem.rem(20)} ${require_rem.rem(25)} ${require_rem.rem(-5)}, rgba(0, 0, 0, 0.04) 0 ${require_rem.rem(10)} ${require_rem.rem(10)} ${require_rem.rem(-5)}`,
		lg: `0 ${require_rem.rem(1)} ${require_rem.rem(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${require_rem.rem(28)} ${require_rem.rem(23)} ${require_rem.rem(-7)}, rgba(0, 0, 0, 0.04) 0 ${require_rem.rem(12)} ${require_rem.rem(12)} ${require_rem.rem(-7)}`,
		xl: `0 ${require_rem.rem(1)} ${require_rem.rem(3)} rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05) 0 ${require_rem.rem(36)} ${require_rem.rem(28)} ${require_rem.rem(-7)}, rgba(0, 0, 0, 0.04) 0 ${require_rem.rem(17)} ${require_rem.rem(17)} ${require_rem.rem(-7)}`
	},
	other: {},
	components: {}
};
//#endregion
exports.DEFAULT_THEME = DEFAULT_THEME;
