// @mantine/core/cjs/core/MantineProvider/Mantine.context.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
let react = require("react");
//#region packages/@mantine/core/src/core/MantineProvider/Mantine.context.ts
const MantineContext = (0, react.createContext)(null);
function useMantineContext() {
	const ctx = (0, react.use)(MantineContext);
	if (!ctx) throw new Error("[@mantine/core] MantineProvider was not found in tree");
	return ctx;
}
function useMantineCssVariablesResolver() {
	return useMantineContext().cssVariablesResolver;
}
function useMantineClassNamesPrefix() {
	return useMantineContext().classNamesPrefix;
}
function useMantineStyleNonce() {
	return useMantineContext().getStyleNonce;
}
function useMantineWithStaticClasses() {
	return useMantineContext().withStaticClasses;
}
function useMantineIsHeadless() {
	return useMantineContext().headless;
}
function useMantineSxTransform() {
	return useMantineContext().stylesTransform?.sx;
}
function useMantineStylesTransform() {
	return useMantineContext().stylesTransform?.styles;
}
function useMantineEnv() {
	return useMantineContext().env || "default";
}
function useMantineDeduplicateInlineStyles() {
	return useMantineContext().deduplicateInlineStyles;
}
//#endregion
exports.MantineContext = MantineContext;
exports.useMantineClassNamesPrefix = useMantineClassNamesPrefix;
exports.useMantineContext = useMantineContext;
exports.useMantineCssVariablesResolver = useMantineCssVariablesResolver;
exports.useMantineDeduplicateInlineStyles = useMantineDeduplicateInlineStyles;
exports.useMantineEnv = useMantineEnv;
exports.useMantineIsHeadless = useMantineIsHeadless;
exports.useMantineStyleNonce = useMantineStyleNonce;
exports.useMantineStylesTransform = useMantineStylesTransform;
exports.useMantineSxTransform = useMantineSxTransform;
exports.useMantineWithStaticClasses = useMantineWithStaticClasses;
