// @mantine/core/esm/components/Code/Code.mjs
"use client";
import { createVarsResolver } from "../../core/styles-api/create-vars-resolver/create-vars-resolver.mjs";
import { getThemeColor } from "../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.mjs";
import { useProps } from "../../core/MantineProvider/use-props/use-props.mjs";
import { useStyles } from "../../core/styles-api/use-styles/use-styles.mjs";
import { factory } from "../../core/factory/factory.mjs";
import { Box } from "../../core/Box/Box.mjs";
import Code_module_default from "./Code.module.mjs";
import { jsx } from "react/jsx-runtime";
//#region packages/@mantine/core/src/components/Code/Code.tsx
const varsResolver = createVarsResolver((theme, { color }) => ({ root: { "--code-bg": color ? getThemeColor(color, theme) : void 0 } }));
const Code = factory((_props) => {
	const props = useProps("Code", null, _props);
	const { classNames, className, style, styles, unstyled, vars, color, block, mod, attributes, ...others } = props;
	const getStyles = useStyles({
		name: "Code",
		props,
		classes: Code_module_default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ jsx(Box, {
		component: block ? "pre" : "code",
		mod: [{ block }, mod],
		...getStyles("root"),
		...others,
		dir: "ltr"
	});
});
Code.classes = Code_module_default;
Code.varsResolver = varsResolver;
Code.displayName = "@mantine/core/Code";
//#endregion
export { Code };
