// @mantine/core/cjs/components/Button/ButtonGroupSection/ButtonGroupSection.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Button_module = require("../Button.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Button/ButtonGroupSection/ButtonGroupSection.tsx
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, gradient, variant, autoContrast, size }) => {
	const colors = theme.variantColorResolver({
		color: color || theme.primaryColor,
		theme,
		gradient,
		variant: variant || "filled",
		autoContrast
	});
	return { groupSection: {
		"--section-height": require_get_size.getSize(size, "section-height"),
		"--section-padding-x": require_get_size.getSize(size, "section-padding-x"),
		"--section-fz": size?.includes("compact") ? require_get_size.getFontSize(size.replace("compact-", "")) : require_get_size.getFontSize(size),
		"--section-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--section-bg": color || variant ? colors.background : void 0,
		"--section-color": colors.color,
		"--section-bd": color || variant ? colors.border : void 0
	} };
});
const ButtonGroupSection = require_factory.factory((_props) => {
	const props = require_use_props.useProps("ButtonGroupSection", null, _props);
	const { className, style, classNames, styles, unstyled, vars, gradient, radius, autoContrast, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...require_use_styles.useStyles({
			name: "ButtonGroupSection",
			props,
			classes: require_Button_module.default,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			vars,
			varsResolver,
			rootSelector: "groupSection"
		})("groupSection"),
		...others
	});
});
ButtonGroupSection.classes = require_Button_module.default;
ButtonGroupSection.varsResolver = varsResolver;
ButtonGroupSection.displayName = "@mantine/core/ButtonGroupSection";
//#endregion
exports.ButtonGroupSection = ButtonGroupSection;
