// @mantine/core/cjs/components/Notification/Notification.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Loader = require("../Loader/Loader.cjs");
const require_CloseButton = require("../CloseButton/CloseButton.cjs");
const require_Notification_module = require("./Notification.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Notification/Notification.tsx
const defaultProps = { withCloseButton: true };
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color }) => ({ root: {
	"--notification-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
	"--notification-color": color ? require_get_theme_color.getThemeColor(color, theme) : void 0
} }));
const Notification = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Notification", defaultProps, _props);
	const { className, color, radius, loading, withCloseButton, withBorder, title, icon, children, onClose, closeButtonProps, classNames, style, styles, unstyled, vars, mod, loaderProps, role, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Notification",
		classes: require_Notification_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		...getStyles("root"),
		mod: [{
			"data-with-icon": !!icon || loading,
			"data-with-border": withBorder
		}, mod],
		role: role || "alert",
		...others,
		children: [
			icon && !loading && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
				...getStyles("icon"),
				children: icon
			}),
			loading && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Loader.Loader, {
				size: 28,
				color,
				...getStyles("loader"),
				...loaderProps
			}),
			/* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
				...getStyles("body"),
				children: [title && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
					...getStyles("title"),
					children: title
				}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
					...getStyles("description"),
					mod: { "data-with-title": !!title },
					children
				})]
			}),
			withCloseButton && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_CloseButton.CloseButton, {
				iconSize: 16,
				color: "gray",
				...closeButtonProps,
				unstyled,
				onClick: (event) => {
					closeButtonProps?.onClick?.(event);
					onClose?.();
				},
				...getStyles("closeButton")
			})
		]
	});
});
Notification.classes = require_Notification_module.default;
Notification.varsResolver = varsResolver;
Notification.displayName = "@mantine/core/Notification";
//#endregion
exports.Notification = Notification;
