// @mantine/core/esm/components/Radio/Radio.mjs
"use client";
import { getRadius, getSize } from "../../core/utils/get-size/get-size.mjs";
import { createVarsResolver } from "../../core/styles-api/create-vars-resolver/create-vars-resolver.mjs";
import { parseThemeColor } from "../../core/MantineProvider/color-functions/parse-theme-color/parse-theme-color.mjs";
import { getThemeColor } from "../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.mjs";
import { getContrastColor } from "../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.mjs";
import { getAutoContrastValue } from "../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.mjs";
import { useProps } from "../../core/MantineProvider/use-props/use-props.mjs";
import { useStyles } from "../../core/styles-api/use-styles/use-styles.mjs";
import { extractStyleProps } from "../../core/Box/style-props/extract-style-props/extract-style-props.mjs";
import { factory } from "../../core/factory/factory.mjs";
import { Box } from "../../core/Box/Box.mjs";
import { InlineInput } from "../../utils/InlineInput/InlineInput.mjs";
import { RadioGroup, RadioGroupContext } from "./RadioGroup/RadioGroup.mjs";
import { RadioCard } from "./RadioCard/RadioCard.mjs";
import { RadioIcon } from "./RadioIcon.mjs";
import { RadioIndicator } from "./RadioIndicator/RadioIndicator.mjs";
import Radio_module_default from "./Radio.module.mjs";
import { use } from "react";
import { useId as useId$1 } from "@mantine/hooks";
import { jsx, jsxs } from "react/jsx-runtime";
//#region packages/@mantine/core/src/components/Radio/Radio.tsx
const defaultProps = {
	labelPosition: "right",
	withErrorStyles: true
};
const varsResolver = createVarsResolver((theme, { size, radius, color, iconColor, variant, autoContrast }) => {
	const parsedColor = parseThemeColor({
		color: color || theme.primaryColor,
		theme
	});
	const outlineColor = parsedColor.isThemeColor && parsedColor.shade === void 0 ? `var(--mantine-color-${parsedColor.color}-outline)` : parsedColor.color;
	return { root: {
		"--radio-size": getSize(size, "radio-size"),
		"--radio-radius": radius === void 0 ? void 0 : getRadius(radius),
		"--radio-color": variant === "outline" ? outlineColor : getThemeColor(color, theme),
		"--radio-icon-color": iconColor ? getThemeColor(iconColor, theme) : getAutoContrastValue(autoContrast, theme) ? getContrastColor({
			color,
			theme,
			autoContrast
		}) : void 0,
		"--radio-icon-size": getSize(size, "radio-icon-size")
	} };
});
const Radio = factory((_props) => {
	const props = useProps("Radio", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, id, size, label, labelPosition, description, error, radius, color, variant, disabled, wrapperProps, icon: Icon = RadioIcon, rootRef, iconColor, onChange, mod, attributes, withErrorStyles, checked, ...others } = props;
	const getStyles = useStyles({
		name: "Radio",
		classes: Radio_module_default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const ctx = use(RadioGroupContext);
	const contextSize = ctx?.size ?? size;
	const componentSize = props.size ? size : contextSize;
	const { styleProps, rest } = extractStyleProps(others);
	const uuid = useId$1(id);
	const describedBy = [
		description ? `${uuid}-description` : void 0,
		error && typeof error !== "boolean" ? `${uuid}-error` : void 0,
		rest["aria-describedby"]
	].filter(Boolean).join(" ") || void 0;
	const contextChecked = ctx ? ctx.value === rest.value : void 0;
	const withContextProps = {
		checked: contextChecked ?? checked,
		name: rest.name ?? ctx?.name,
		onChange: (event) => {
			ctx?.onChange(event);
			onChange?.(event);
		},
		disabled: ctx?.disabled ?? disabled
	};
	return /* @__PURE__ */ jsx(InlineInput, {
		...getStyles("root"),
		__staticSelector: "Radio",
		__stylesApiProps: props,
		id: uuid,
		size: componentSize,
		labelPosition,
		label,
		description,
		error,
		disabled: withContextProps.disabled,
		classNames,
		styles,
		unstyled,
		"data-checked": (contextChecked ?? checked) || void 0,
		variant,
		ref: rootRef,
		mod,
		attributes,
		...styleProps,
		...wrapperProps,
		children: /* @__PURE__ */ jsxs(Box, {
			...getStyles("inner"),
			mod: { "label-position": labelPosition },
			children: [/* @__PURE__ */ jsx(Box, {
				...getStyles("radio", {
					focusable: true,
					variant
				}),
				...rest,
				...withContextProps,
				component: "input",
				mod: {
					error: !!error,
					"with-error-styles": withErrorStyles
				},
				id: uuid,
				type: "radio",
				"aria-describedby": describedBy
			}), /* @__PURE__ */ jsx(Icon, {
				...getStyles("icon"),
				"aria-hidden": true
			})]
		})
	});
});
Radio.classes = Radio_module_default;
Radio.varsResolver = varsResolver;
Radio.displayName = "@mantine/core/Radio";
Radio.Group = RadioGroup;
Radio.Card = RadioCard;
Radio.Indicator = RadioIndicator;
//#endregion
export { Radio };
