// @mantine/core/cjs/components/Input/InputClearButton/InputClearButton.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_resolved_styles_api = require("../../../core/styles-api/use-resolved-styles-api/use-resolved-styles-api.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_CloseButton = require("../../CloseButton/CloseButton.cjs");
const require_Input_context = require("../Input.context.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Input/InputClearButton/InputClearButton.tsx
const InputClearButton = require_factory.factory((_props) => {
	const props = require_use_props.useProps("InputClearButton", null, _props);
	const { size, variant, vars, classNames, styles, ...others } = props;
	const ctx = (0, react.use)(require_Input_context.InputContext);
	const { resolvedClassNames, resolvedStyles } = require_use_resolved_styles_api.useResolvedStylesApi({
		classNames,
		styles,
		props
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_CloseButton.CloseButton, {
		variant: variant || "transparent",
		size: size || ctx?.size || "sm",
		classNames: resolvedClassNames,
		styles: resolvedStyles,
		__staticSelector: "InputClearButton",
		style: {
			pointerEvents: "all",
			background: "var(--input-bg)",
			...others.style
		},
		...others
	});
});
InputClearButton.displayName = "@mantine/core/InputClearButton";
//#endregion
exports.InputClearButton = InputClearButton;
