// @mantine/core/cjs/components/Alert/Alert.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_CloseButton = require("../CloseButton/CloseButton.cjs");
const require_Alert_module = require("./Alert.module.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Alert/Alert.tsx
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, variant, autoContrast }) => {
	const colors = theme.variantColorResolver({
		color: color || theme.primaryColor,
		theme,
		variant: variant || "light",
		autoContrast
	});
	return { root: {
		"--alert-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--alert-bg": color || variant ? colors.background : void 0,
		"--alert-color": colors.color,
		"--alert-bd": color || variant ? colors.border : void 0
	} };
});
const Alert = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Alert", null, _props);
	const { classNames, className, style, styles, unstyled, vars, radius, color, title, children, id, icon, withCloseButton, onClose, closeButtonLabel, variant, autoContrast, role, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Alert",
		classes: require_Alert_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const rootId = (0, _mantine_hooks.useId)(id);
	const titleId = title && `${rootId}-title` || void 0;
	const bodyId = `${rootId}-body`;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		id: rootId,
		...getStyles("root", { variant }),
		variant,
		...others,
		role: role || "alert",
		"aria-describedby": children ? bodyId : void 0,
		"aria-labelledby": title ? titleId : void 0,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
			...getStyles("wrapper"),
			children: [
				icon && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
					...getStyles("icon"),
					children: icon
				}),
				/* @__PURE__ */ (0, react_jsx_runtime.jsxs)("div", {
					...getStyles("body"),
					children: [title && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
						...getStyles("title"),
						"data-with-close-button": withCloseButton || void 0,
						children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
							id: titleId,
							...getStyles("label"),
							children: title
						})
					}), children && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", {
						id: bodyId,
						...getStyles("message"),
						"data-variant": variant,
						children
					})]
				}),
				withCloseButton && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_CloseButton.CloseButton, {
					...getStyles("closeButton"),
					onClick: onClose,
					variant: "transparent",
					size: 16,
					iconSize: 16,
					"aria-label": closeButtonLabel,
					unstyled
				})
			]
		})
	});
});
Alert.classes = require_Alert_module.default;
Alert.varsResolver = varsResolver;
Alert.displayName = "@mantine/core/Alert";
//#endregion
exports.Alert = Alert;
