// @mantine/core/cjs/components/Combobox/ComboboxSearch/ComboboxSearch.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Input = require("../../Input/Input.cjs");
const require_Combobox_module = require("../Combobox.module.cjs");
const require_Combobox_context = require("../Combobox.context.cjs");
const require_use_combobox_target_props = require("../use-combobox-target-props/use-combobox-target-props.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Combobox/ComboboxSearch/ComboboxSearch.tsx
const defaultProps = {
	withAriaAttributes: true,
	withKeyboardNavigation: true
};
const ComboboxSearch = require_factory.factory((_props) => {
	const { classNames, styles, unstyled, vars, withAriaAttributes, onKeyDown, onClick, withKeyboardNavigation, size, ref, ...others } = require_use_props.useProps("ComboboxSearch", defaultProps, _props);
	const ctx = require_Combobox_context.useComboboxContext();
	const _styles = ctx.getStyles("search");
	const targetProps = require_use_combobox_target_props.useComboboxTargetProps({
		targetType: "input",
		withAriaAttributes,
		withKeyboardNavigation,
		withExpandedAttribute: false,
		onKeyDown,
		onClick,
		autoComplete: "off"
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Input.Input, {
		ref: (0, _mantine_hooks.useMergedRef)(ref, ctx.store.searchRef),
		classNames: [{ input: _styles.className }, classNames],
		styles: [{ input: _styles.style }, styles],
		size: size || ctx.size,
		...targetProps,
		...others,
		__staticSelector: "Combobox"
	});
});
ComboboxSearch.classes = require_Combobox_module.default;
ComboboxSearch.displayName = "@mantine/core/ComboboxSearch";
//#endregion
exports.ComboboxSearch = ComboboxSearch;
