// @mantine/core/cjs/components/ColorPicker/Saturation/Saturation.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_ColorPicker_context = require("../ColorPicker.context.cjs");
const require_Thumb = require("../Thumb/Thumb.cjs");
const require_converters = require("../converters/converters.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/ColorPicker/Saturation/Saturation.tsx
function Saturation({ className, onChange, onChangeEnd, value, saturationLabel, focusable = true, size, color, onScrubStart, onScrubEnd, ...others }) {
	const { getStyles } = (0, react.use)(require_ColorPicker_context.ColorPickerContext);
	const [position, setPosition] = (0, react.useState)({
		x: value.s / 100,
		y: 1 - value.v / 100
	});
	const positionRef = (0, react.useRef)(position);
	const { ref } = (0, _mantine_hooks.useMove)(({ x, y }) => {
		positionRef.current = {
			x,
			y
		};
		onChange({
			s: Math.round(x * 100),
			v: Math.round((1 - y) * 100)
		});
	}, {
		onScrubEnd: () => {
			const { x, y } = positionRef.current;
			onChangeEnd({
				s: Math.round(x * 100),
				v: Math.round((1 - y) * 100)
			});
			onScrubEnd?.();
		},
		onScrubStart
	});
	(0, react.useEffect)(() => {
		setPosition({
			x: value.s / 100,
			y: 1 - value.v / 100
		});
	}, [value.s, value.v]);
	const handleArrow = (event, pos) => {
		event.preventDefault();
		const _position = (0, _mantine_hooks.clampUseMovePosition)(pos);
		onChange({
			s: Math.round(_position.x * 100),
			v: Math.round((1 - _position.y) * 100)
		});
		onChangeEnd({
			s: Math.round(_position.x * 100),
			v: Math.round((1 - _position.y) * 100)
		});
	};
	const handleKeyDown = (event) => {
		switch (event.key) {
			case "ArrowUp":
				handleArrow(event, {
					y: position.y - .05,
					x: position.x
				});
				break;
			case "ArrowDown":
				handleArrow(event, {
					y: position.y + .05,
					x: position.x
				});
				break;
			case "ArrowRight":
				handleArrow(event, {
					x: position.x + .05,
					y: position.y
				});
				break;
			case "ArrowLeft":
				handleArrow(event, {
					x: position.x - .05,
					y: position.y
				});
				break;
		}
	};
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		...getStyles("saturation"),
		ref,
		...others,
		role: "slider",
		"aria-label": saturationLabel,
		"aria-valuenow": position.x,
		"aria-valuetext": require_converters.convertHsvaTo("rgba", value),
		tabIndex: focusable ? 0 : -1,
		onKeyDown: handleKeyDown,
		children: [
			/* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", { ...getStyles("saturationOverlay", { style: { backgroundColor: `hsl(${value.h}, 100%, 50%)` } }) }),
			/* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", { ...getStyles("saturationOverlay", { style: { backgroundImage: "linear-gradient(90deg, #fff, transparent)" } }) }),
			/* @__PURE__ */ (0, react_jsx_runtime.jsx)("div", { ...getStyles("saturationOverlay", { style: { backgroundImage: "linear-gradient(0deg, #000, transparent)" } }) }),
			/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Thumb.Thumb, {
				position,
				...getStyles("thumb", { style: { backgroundColor: color } })
			})
		]
	});
}
Saturation.displayName = "@mantine/core/Saturation";
//#endregion
exports.Saturation = Saturation;
