// @mantine/core/cjs/components/Blockquote/Blockquote.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../core/utils/units-converters/rem.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_parse_theme_color = require("../../core/MantineProvider/color-functions/parse-theme-color/parse-theme-color.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_rgba = require("../../core/MantineProvider/color-functions/rgba/rgba.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Blockquote_module = require("./Blockquote.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Blockquote/Blockquote.tsx
const defaultProps = { iconSize: 48 };
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { color, iconSize, radius }) => {
	const darkParsed = require_parse_theme_color.parseThemeColor({
		color: color || theme.primaryColor,
		theme,
		colorScheme: "dark"
	});
	return { root: {
		"--bq-bg-light": require_rgba.rgba(require_parse_theme_color.parseThemeColor({
			color: color || theme.primaryColor,
			theme,
			colorScheme: "light"
		}).value, .07),
		"--bq-bg-dark": require_rgba.rgba(darkParsed.value, .06),
		"--bq-bd": require_get_theme_color.getThemeColor(color, theme),
		"--bq-icon-size": require_rem.rem(iconSize),
		"--bq-radius": require_get_size.getRadius(radius)
	} };
});
const Blockquote = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Blockquote", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, children, icon, iconSize, cite, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Blockquote",
		classes: require_Blockquote_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		component: "blockquote",
		...getStyles("root"),
		...others,
		children: [
			icon && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("span", {
				...getStyles("icon"),
				children: icon
			}),
			children,
			cite && /* @__PURE__ */ (0, react_jsx_runtime.jsx)("cite", {
				...getStyles("cite"),
				children: cite
			})
		]
	});
});
Blockquote.classes = require_Blockquote_module.default;
Blockquote.varsResolver = varsResolver;
Blockquote.displayName = "@mantine/core/Blockquote";
//#endregion
exports.Blockquote = Blockquote;
