// @mantine/core/cjs/core/MantineProvider/MantineCssVariables/MantineCssVariables.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_Mantine_context = require("../Mantine.context.cjs");
const require_MantineThemeProvider = require("../MantineThemeProvider/MantineThemeProvider.cjs");
const require_convert_css_variables = require("../convert-css-variables/convert-css-variables.cjs");
const require_get_merged_variables = require("./get-merged-variables.cjs");
const require_remove_default_variables = require("./remove-default-variables.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/core/MantineProvider/MantineCssVariables/MantineCssVariables.tsx
function getColorSchemeCssVariables(selectorOverride) {
	return require_convert_css_variables.convertCssVariables({
		variables: {},
		dark: { "--mantine-color-scheme": "dark" },
		light: { "--mantine-color-scheme": "light" }
	}, selectorOverride);
}
function MantineCssVariables({ cssVariablesSelector, deduplicateCssVariables }) {
	const theme = require_MantineThemeProvider.useMantineTheme();
	const nonce = require_Mantine_context.useMantineStyleNonce();
	const mergedVariables = require_get_merged_variables.getMergedVariables({
		theme,
		generator: require_Mantine_context.useMantineCssVariablesResolver()
	});
	const shouldCleanVariables = (cssVariablesSelector === void 0 || cssVariablesSelector === ":root" || cssVariablesSelector === ":host") && deduplicateCssVariables;
	const css = require_convert_css_variables.convertCssVariables(shouldCleanVariables ? require_remove_default_variables.removeDefaultVariables(mergedVariables) : mergedVariables, cssVariablesSelector);
	if (css) return /* @__PURE__ */ (0, react_jsx_runtime.jsx)("style", {
		"data-mantine-styles": true,
		nonce: nonce?.(),
		dangerouslySetInnerHTML: { __html: `${css}${shouldCleanVariables ? "" : getColorSchemeCssVariables(cssVariablesSelector)}` }
	});
	return null;
}
MantineCssVariables.displayName = "@mantine/CssVariables";
//#endregion
exports.MantineCssVariables = MantineCssVariables;
