// @mantine/core/cjs/components/Input/InputPlaceholder/InputPlaceholder.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Input_module = require("../Input.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Input/InputPlaceholder/InputPlaceholder.tsx
const InputPlaceholder = require_factory.factory((_props) => {
	const props = require_use_props.useProps("InputPlaceholder", null, _props);
	const { classNames, className, style, styles, unstyled, vars, __staticSelector, error, mod, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...require_use_styles.useStyles({
			name: ["InputPlaceholder", __staticSelector],
			props,
			classes: require_Input_module.default,
			className,
			style,
			classNames,
			styles,
			unstyled,
			attributes,
			rootSelector: "placeholder"
		})("placeholder"),
		mod: [{ error: !!error }, mod],
		component: "span",
		...others
	});
});
InputPlaceholder.classes = require_Input_module.default;
InputPlaceholder.displayName = "@mantine/core/InputPlaceholder";
//#endregion
exports.InputPlaceholder = InputPlaceholder;
