// @mantine/core/cjs/components/Checkbox/Checkbox.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_parse_theme_color = require("../../core/MantineProvider/color-functions/parse-theme-color/parse-theme-color.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_get_contrast_color = require("../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs");
const require_get_auto_contrast_value = require("../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_extract_style_props = require("../../core/Box/style-props/extract-style-props/extract-style-props.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_InlineInput = require("../../utils/InlineInput/InlineInput.cjs");
const require_CheckboxGroup = require("./CheckboxGroup/CheckboxGroup.cjs");
const require_CheckboxCard = require("./CheckboxCard/CheckboxCard.cjs");
const require_CheckIcon = require("./CheckIcon.cjs");
const require_CheckboxIndicator = require("./CheckboxIndicator/CheckboxIndicator.cjs");
const require_Checkbox_module = require("./Checkbox.module.cjs");
let react = require("react");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Checkbox/Checkbox.tsx
const defaultProps = {
	labelPosition: "right",
	icon: require_CheckIcon.CheckboxIcon,
	withErrorStyles: true,
	variant: "filled",
	radius: "sm"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, size, iconColor, variant, autoContrast }) => {
	const parsedColor = require_parse_theme_color.parseThemeColor({
		color: color || theme.primaryColor,
		theme
	});
	const outlineColor = parsedColor.isThemeColor && parsedColor.shade === void 0 ? `var(--mantine-color-${parsedColor.color}-outline)` : parsedColor.color;
	return { root: {
		"--checkbox-size": require_get_size.getSize(size, "checkbox-size"),
		"--checkbox-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--checkbox-color": variant === "outline" ? outlineColor : require_get_theme_color.getThemeColor(color, theme),
		"--checkbox-icon-color": iconColor ? require_get_theme_color.getThemeColor(iconColor, theme) : require_get_auto_contrast_value.getAutoContrastValue(autoContrast, theme) ? require_get_contrast_color.getContrastColor({
			color,
			theme,
			autoContrast
		}) : void 0
	} };
});
const Checkbox = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Checkbox", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, color, label, id, size, radius, wrapperProps, checked, labelPosition, description, error, disabled, variant, indeterminate, icon: Icon, rootRef, iconColor, onChange, autoContrast, mod, attributes, readOnly, onClick, withErrorStyles, ref, ...others } = props;
	const inputRef = (0, react.useRef)(null);
	const ctx = (0, react.use)(require_CheckboxGroup.CheckboxGroupContext);
	const _size = size || ctx?.size;
	const getStyles = require_use_styles.useStyles({
		name: "Checkbox",
		props,
		classes: require_Checkbox_module.default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const { styleProps, rest } = require_extract_style_props.extractStyleProps(others);
	const uuid = (0, _mantine_hooks.useId)(id);
	const describedBy = [
		description ? `${uuid}-description` : void 0,
		error && typeof error !== "boolean" ? `${uuid}-error` : void 0,
		rest["aria-describedby"]
	].filter(Boolean).join(" ") || void 0;
	const withContextProps = {
		checked: ctx?.value.includes(rest.value) ?? checked,
		onChange: (event) => {
			ctx?.onChange(event);
			onChange?.(event);
		}
	};
	const isDisabledByGroup = ctx?.isDisabled?.(rest.value) ?? false;
	const finalDisabled = disabled || isDisabledByGroup;
	(0, react.useEffect)(() => {
		if (inputRef.current) {
			inputRef.current.indeterminate = indeterminate || false;
			if (indeterminate) inputRef.current.setAttribute("data-indeterminate", "true");
			else inputRef.current.removeAttribute("data-indeterminate");
		}
	}, [indeterminate]);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_InlineInput.InlineInput, {
		...getStyles("root"),
		__staticSelector: "Checkbox",
		__stylesApiProps: props,
		id: uuid,
		size: _size,
		labelPosition,
		label,
		description,
		error,
		disabled: finalDisabled,
		classNames,
		styles,
		unstyled,
		"data-checked": withContextProps.checked || checked || void 0,
		variant,
		ref: rootRef,
		mod,
		attributes,
		inert: rest.inert,
		...styleProps,
		...wrapperProps,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
			...getStyles("inner"),
			mod: { "data-label-position": labelPosition },
			children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
				component: "input",
				id: uuid,
				ref: (0, _mantine_hooks.useMergedRef)(inputRef, ref),
				mod: {
					error: !!error,
					"with-error-styles": withErrorStyles
				},
				...getStyles("input", {
					focusable: true,
					variant
				}),
				...rest,
				...withContextProps,
				"aria-describedby": describedBy,
				disabled: finalDisabled,
				inert: rest.inert,
				type: "checkbox",
				onClick: (event) => {
					if (readOnly) event.preventDefault();
					onClick?.(event);
				}
			}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(Icon, {
				indeterminate,
				...getStyles("icon")
			})]
		})
	});
});
Checkbox.classes = {
	...require_Checkbox_module.default,
	...require_InlineInput.InlineInputClasses
};
Checkbox.varsResolver = varsResolver;
Checkbox.displayName = "@mantine/core/Checkbox";
Checkbox.Group = require_CheckboxGroup.CheckboxGroup;
Checkbox.Indicator = require_CheckboxIndicator.CheckboxIndicator;
Checkbox.Card = require_CheckboxCard.CheckboxCard;
//#endregion
exports.Checkbox = Checkbox;
