// @mantine/core/esm/components/Tabs/Tabs.mjs
"use client";
import { getSafeId } from "../../core/utils/get-safe-id/get-safe-id.mjs";
import { getRadius } from "../../core/utils/get-size/get-size.mjs";
import { createVarsResolver } from "../../core/styles-api/create-vars-resolver/create-vars-resolver.mjs";
import { getThemeColor } from "../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.mjs";
import { getContrastColor } from "../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.mjs";
import { getAutoContrastValue } from "../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.mjs";
import { useProps } from "../../core/MantineProvider/use-props/use-props.mjs";
import { useStyles } from "../../core/styles-api/use-styles/use-styles.mjs";
import { factory } from "../../core/factory/factory.mjs";
import { Box } from "../../core/Box/Box.mjs";
import { TabsProvider } from "./Tabs.context.mjs";
import Tabs_module_default from "./Tabs.module.mjs";
import { TabsList } from "./TabsList/TabsList.mjs";
import { TabsPanel } from "./TabsPanel/TabsPanel.mjs";
import { TabsTab } from "./TabsTab/TabsTab.mjs";
import { useId, useUncontrolled } from "@mantine/hooks";
import { jsx } from "react/jsx-runtime";
//#region packages/@mantine/core/src/components/Tabs/Tabs.tsx
const VALUE_ERROR = "Tabs.Tab or Tabs.Panel component was rendered with invalid value or without value";
const defaultProps = {
	keepMounted: true,
	keepMountedMode: "activity",
	orientation: "horizontal",
	loop: true,
	activateTabWithKeyboard: true,
	variant: "default",
	placement: "left"
};
const varsResolver = createVarsResolver((theme, { radius, color, autoContrast }) => ({ root: {
	"--tabs-radius": getRadius(radius),
	"--tabs-color": getThemeColor(color, theme),
	"--tabs-text-color": getAutoContrastValue(autoContrast, theme) ? getContrastColor({
		color,
		theme,
		autoContrast
	}) : void 0
} }));
const Tabs = factory((_props) => {
	const props = useProps("Tabs", defaultProps, _props);
	const { defaultValue, value, onChange, orientation, children, loop, id, activateTabWithKeyboard, allowTabDeactivation, variant, color, radius, inverted, placement, keepMounted, keepMountedMode, classNames, styles, unstyled, className, style, vars, autoContrast, mod, attributes, ...others } = props;
	const uid = useId(id);
	const [currentTab, setCurrentTab] = useUncontrolled({
		value,
		defaultValue,
		finalValue: null,
		onChange
	});
	const getStyles = useStyles({
		name: "Tabs",
		props,
		classes: Tabs_module_default,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ jsx(TabsProvider, {
		value: {
			placement,
			value: currentTab,
			orientation,
			id: uid,
			loop,
			activateTabWithKeyboard,
			getTabId: getSafeId(`${uid}-tab`, VALUE_ERROR),
			getPanelId: getSafeId(`${uid}-panel`, VALUE_ERROR),
			onChange: setCurrentTab,
			allowTabDeactivation,
			variant,
			color,
			radius,
			inverted,
			keepMounted,
			keepMountedMode,
			unstyled,
			getStyles
		},
		children: /* @__PURE__ */ jsx(Box, {
			id: uid,
			variant,
			mod: [{
				orientation,
				inverted: orientation === "horizontal" && inverted,
				placement: orientation === "vertical" && placement
			}, mod],
			...getStyles("root"),
			...others,
			children
		})
	});
});
Tabs.classes = Tabs_module_default;
Tabs.varsResolver = varsResolver;
Tabs.displayName = "@mantine/core/Tabs";
Tabs.Tab = TabsTab;
Tabs.Panel = TabsPanel;
Tabs.List = TabsList;
//#endregion
export { Tabs };
