// @mantine/core/cjs/components/AppShell/AppShell.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_default_z_index = require("../../core/utils/get-default-z-index/get-default-z-index.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_AppShell_context = require("./AppShell.context.cjs");
const require_AppShell_module = require("./AppShell.module.cjs");
const require_AppShellAside = require("./AppShellAside/AppShellAside.cjs");
const require_AppShellFooter = require("./AppShellFooter/AppShellFooter.cjs");
const require_AppShellHeader = require("./AppShellHeader/AppShellHeader.cjs");
const require_AppShellMain = require("./AppShellMain/AppShellMain.cjs");
const require_AppShellMediaStyles = require("./AppShellMediaStyles/AppShellMediaStyles.cjs");
const require_AppShellNavbar = require("./AppShellNavbar/AppShellNavbar.cjs");
const require_AppShellSection = require("./AppShellSection/AppShellSection.cjs");
const require_use_resizing = require("./use-resizing/use-resizing.cjs");
let _mantine_hooks = require("@mantine/hooks");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/AppShell/AppShell.tsx
const defaultProps = {
	withBorder: true,
	padding: 0,
	transitionDuration: 200,
	transitionTimingFunction: "ease",
	zIndex: require_get_default_z_index.getDefaultZIndex("app"),
	mode: "fixed"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { transitionDuration, transitionTimingFunction }) => ({ root: {
	"--app-shell-transition-duration": `${transitionDuration}ms`,
	"--app-shell-transition-timing-function": transitionTimingFunction
} }));
const AppShell = require_factory.factory((_props) => {
	const props = require_use_props.useProps("AppShell", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, navbar, withBorder, padding, transitionDuration, transitionTimingFunction, header, zIndex, layout, disabled, aside, footer, offsetScrollbars = true, mode, mod, attributes, id, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "AppShell",
		classes: require_AppShell_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const resizing = require_use_resizing.useResizing({
		disabled,
		transitionDuration
	});
	const _id = (0, _mantine_hooks.useId)(id);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_AppShell_context.AppShellProvider, {
		value: {
			getStyles,
			withBorder,
			zIndex,
			disabled,
			offsetScrollbars,
			mode
		},
		children: [/* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_AppShellMediaStyles.AppShellMediaStyles, {
			navbar,
			header,
			aside,
			footer,
			padding,
			mode,
			selector: mode === "static" ? `#${_id}` : void 0
		}), /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			...getStyles("root"),
			id: _id,
			mod: [{
				resizing,
				layout,
				disabled,
				mode
			}, mod],
			...others
		})]
	});
});
AppShell.classes = require_AppShell_module.default;
AppShell.varsResolver = varsResolver;
AppShell.displayName = "@mantine/core/AppShell";
AppShell.Navbar = require_AppShellNavbar.AppShellNavbar;
AppShell.Header = require_AppShellHeader.AppShellHeader;
AppShell.Main = require_AppShellMain.AppShellMain;
AppShell.Aside = require_AppShellAside.AppShellAside;
AppShell.Footer = require_AppShellFooter.AppShellFooter;
AppShell.Section = require_AppShellSection.AppShellSection;
//#endregion
exports.AppShell = AppShell;
