// @mantine/core/cjs/components/Radio/RadioCard/RadioCard.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_DirectionProvider = require("../../../core/DirectionProvider/DirectionProvider.cjs");
const require_UnstyledButton = require("../../UnstyledButton/UnstyledButton.cjs");
const require_RadioGroup = require("../RadioGroup/RadioGroup.cjs");
const require_RadioCard_module = require("./RadioCard.module.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Radio/RadioCard/RadioCard.tsx
const RadioCardContext = (0, react.createContext)(null);
const defaultProps = { withBorder: true };
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { radius }) => ({ card: { "--card-radius": require_get_size.getRadius(radius) } }));
const RadioCard = require_factory.factory((_props) => {
	const props = require_use_props.useProps("RadioCard", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, checked, mod, withBorder, value, onClick, name, onKeyDown, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "RadioCard",
		classes: require_RadioCard_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver,
		rootSelector: "card"
	});
	const { dir } = require_DirectionProvider.useDirection();
	const ctx = (0, react.use)(require_RadioGroup.RadioGroupContext);
	const _checked = typeof checked === "boolean" ? checked : ctx?.value === value || false;
	const _name = name || ctx?.name;
	const handleKeyDown = (event) => {
		onKeyDown?.(event);
		if (!_name) return;
		if ([
			"ArrowDown",
			"ArrowUp",
			"ArrowLeft",
			"ArrowRight"
		].includes(event.nativeEvent.code)) {
			event.preventDefault();
			const siblings = Array.from(document.querySelectorAll(`[role="radio"][name="${_name}"]`));
			const currentIndex = siblings.findIndex((element) => element === event.target);
			const nextIndex = currentIndex + 1 >= siblings.length ? 0 : currentIndex + 1;
			const prevIndex = currentIndex - 1 < 0 ? siblings.length - 1 : currentIndex - 1;
			if (event.nativeEvent.code === "ArrowDown") {
				siblings[nextIndex].focus();
				siblings[nextIndex].click();
			}
			if (event.nativeEvent.code === "ArrowUp") {
				siblings[prevIndex].focus();
				siblings[prevIndex].click();
			}
			if (event.nativeEvent.code === "ArrowLeft") {
				siblings[dir === "ltr" ? prevIndex : nextIndex].focus();
				siblings[dir === "ltr" ? prevIndex : nextIndex].click();
			}
			if (event.nativeEvent.code === "ArrowRight") {
				siblings[dir === "ltr" ? nextIndex : prevIndex].focus();
				siblings[dir === "ltr" ? nextIndex : prevIndex].click();
			}
		}
	};
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(RadioCardContext, {
		value: { checked: _checked },
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_UnstyledButton.UnstyledButton, {
			mod: [{
				"with-border": withBorder,
				checked: _checked
			}, mod],
			...getStyles("card"),
			...others,
			role: "radio",
			"aria-checked": _checked,
			name: _name,
			onClick: (event) => {
				onClick?.(event);
				ctx?.onChange(value || "");
			},
			onKeyDown: handleKeyDown
		})
	});
});
RadioCard.displayName = "@mantine/core/RadioCard";
RadioCard.classes = require_RadioCard_module.default;
RadioCard.varsResolver = varsResolver;
//#endregion
exports.RadioCard = RadioCard;
exports.RadioCardContext = RadioCardContext;
