// @mantine/core/cjs/core/utils/get-size/get-size.cjs
"use client";
const require_rem = require("../units-converters/rem.cjs");
const require_is_number_like = require("../is-number-like/is-number-like.cjs");
//#region packages/@mantine/core/src/core/utils/get-size/get-size.ts
function getSize(size, prefix = "size", convertToRem = true) {
	if (size === void 0) return;
	return require_is_number_like.isNumberLike(size) ? convertToRem ? require_rem.rem(size) : size : `var(--${prefix}-${size})`;
}
function getSpacing(size) {
	return getSize(size, "mantine-spacing");
}
function getRadius(size) {
	if (size === void 0) return "var(--mantine-radius-default)";
	return getSize(size, "mantine-radius");
}
function getFontSize(size) {
	return getSize(size, "mantine-font-size");
}
function getLineHeight(size) {
	return getSize(size, "mantine-line-height", false);
}
function getShadow(size) {
	if (!size) return;
	return getSize(size, "mantine-shadow", false);
}
//#endregion
exports.getFontSize = getFontSize;
exports.getLineHeight = getLineHeight;
exports.getRadius = getRadius;
exports.getShadow = getShadow;
exports.getSize = getSize;
exports.getSpacing = getSpacing;
