// @mantine/core/cjs/components/Avatar/AvatarGroup/AvatarGroup.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_Avatar_module = require("../Avatar.module.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Avatar/AvatarGroup/AvatarGroup.tsx
const AvatarGroupContext = (0, react.createContext)({ withinGroup: false });
const varsResolver = require_create_vars_resolver.createVarsResolver((_, { spacing }) => ({ group: { "--ag-spacing": require_get_size.getSpacing(spacing) } }));
const AvatarGroup = require_factory.factory((_props) => {
	const props = require_use_props.useProps("AvatarGroup", null, _props);
	const { classNames, className, style, styles, unstyled, vars, spacing, attributes, ...others } = props;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(AvatarGroupContext, {
		value: { withinGroup: true },
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			...require_use_styles.useStyles({
				name: "AvatarGroup",
				classes: require_Avatar_module.default,
				props,
				className,
				style,
				classNames,
				styles,
				unstyled,
				attributes,
				vars,
				varsResolver,
				rootSelector: "group"
			})("group"),
			...others
		})
	});
});
AvatarGroup.classes = require_Avatar_module.default;
AvatarGroup.varsResolver = varsResolver;
AvatarGroup.displayName = "@mantine/core/AvatarGroup";
//#endregion
exports.AvatarGroup = AvatarGroup;
exports.AvatarGroupContext = AvatarGroupContext;
