// @mantine/core/cjs/components/Divider/Divider.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_Divider_module = require("./Divider.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Divider/Divider.tsx
const defaultProps = { orientation: "horizontal" };
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { color, variant, size }) => ({ root: {
	"--divider-color": color ? require_get_theme_color.getThemeColor(color, theme) : void 0,
	"--divider-border-style": variant,
	"--divider-size": require_get_size.getSize(size, "divider-size")
} }));
const Divider = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Divider", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, color, orientation, label, labelPosition, mod, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Divider",
		classes: require_Divider_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		mod: [{
			orientation,
			withLabel: !!label
		}, mod],
		role: "separator",
		...getStyles("root"),
		...others,
		children: label && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			component: "span",
			mod: { position: labelPosition },
			...getStyles("label"),
			children: label
		})
	});
});
Divider.classes = require_Divider_module.default;
Divider.varsResolver = varsResolver;
Divider.displayName = "@mantine/core/Divider";
//#endregion
exports.Divider = Divider;
