// @mantine/core/cjs/components/Indicator/Indicator.cjs
"use client";
require("../../_virtual/_rolldown/runtime.cjs");
const require_rem = require("../../core/utils/units-converters/rem.cjs");
const require_get_size = require("../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_get_theme_color = require("../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_get_contrast_color = require("../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs");
const require_get_auto_contrast_value = require("../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.cjs");
const require_use_props = require("../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../core/factory/factory.cjs");
const require_Box = require("../../core/Box/Box.cjs");
const require_get_position_variables = require("./get-position-variables/get-position-variables.cjs");
const require_Indicator_module = require("./Indicator.module.cjs");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Indicator/Indicator.tsx
const defaultProps = {
	position: "top-end",
	offset: 0,
	showZero: true
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { color, position, offset, size, radius, zIndex, autoContrast }) => ({ root: {
	"--indicator-color": color ? require_get_theme_color.getThemeColor(color, theme) : void 0,
	"--indicator-text-color": require_get_auto_contrast_value.getAutoContrastValue(autoContrast, theme) ? require_get_contrast_color.getContrastColor({
		color,
		theme,
		autoContrast
	}) : void 0,
	"--indicator-size": require_rem.rem(size),
	"--indicator-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
	"--indicator-z-index": zIndex?.toString(),
	...require_get_position_variables.getPositionVariables(position, offset)
} }));
const Indicator = require_factory.factory((_props) => {
	const props = require_use_props.useProps("Indicator", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, children, position, offset, inline, label, radius, color, withBorder, disabled, processing, zIndex, autoContrast, maxValue, showZero, mod, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "Indicator",
		classes: require_Indicator_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver
	});
	const shouldHideZero = !showZero && (label === 0 || label === "0");
	const formattedLabel = maxValue !== void 0 && typeof label === "number" && label > maxValue ? `${maxValue}+` : label;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsxs)(require_Box.Box, {
		...getStyles("root"),
		mod: [{ inline }, mod],
		...others,
		children: [!disabled && !shouldHideZero && /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
			mod: {
				"with-label": !!label,
				"with-border": withBorder,
				processing
			},
			...getStyles("indicator"),
			children: formattedLabel
		}), children]
	});
});
Indicator.classes = require_Indicator_module.default;
Indicator.varsResolver = varsResolver;
Indicator.displayName = "@mantine/core/Indicator";
//#endregion
exports.Indicator = Indicator;
