// @mantine/core/cjs/components/Checkbox/CheckboxIndicator/CheckboxIndicator.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_get_size = require("../../../core/utils/get-size/get-size.cjs");
const require_create_vars_resolver = require("../../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs");
const require_parse_theme_color = require("../../../core/MantineProvider/color-functions/parse-theme-color/parse-theme-color.cjs");
const require_get_theme_color = require("../../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs");
const require_get_contrast_color = require("../../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs");
const require_get_auto_contrast_value = require("../../../core/MantineProvider/color-functions/get-auto-contrast-value/get-auto-contrast-value.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
const require_use_styles = require("../../../core/styles-api/use-styles/use-styles.cjs");
const require_factory = require("../../../core/factory/factory.cjs");
const require_Box = require("../../../core/Box/Box.cjs");
const require_CheckboxCard = require("../CheckboxCard/CheckboxCard.cjs");
const require_CheckIcon = require("../CheckIcon.cjs");
const require_CheckboxIndicator_module = require("./CheckboxIndicator.module.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
//#region packages/@mantine/core/src/components/Checkbox/CheckboxIndicator/CheckboxIndicator.tsx
const defaultProps = {
	icon: require_CheckIcon.CheckboxIcon,
	variant: "filled",
	radius: "sm"
};
const varsResolver = require_create_vars_resolver.createVarsResolver((theme, { radius, color, size, iconColor, variant, autoContrast }) => {
	const parsedColor = require_parse_theme_color.parseThemeColor({
		color: color || theme.primaryColor,
		theme
	});
	const outlineColor = parsedColor.isThemeColor && parsedColor.shade === void 0 ? `var(--mantine-color-${parsedColor.color}-outline)` : parsedColor.color;
	return { indicator: {
		"--checkbox-size": require_get_size.getSize(size, "checkbox-size"),
		"--checkbox-radius": radius === void 0 ? void 0 : require_get_size.getRadius(radius),
		"--checkbox-color": variant === "outline" ? outlineColor : require_get_theme_color.getThemeColor(color, theme),
		"--checkbox-icon-color": iconColor ? require_get_theme_color.getThemeColor(iconColor, theme) : require_get_auto_contrast_value.getAutoContrastValue(autoContrast, theme) ? require_get_contrast_color.getContrastColor({
			color,
			theme,
			autoContrast
		}) : void 0
	} };
});
const CheckboxIndicator = require_factory.factory((_props) => {
	const props = require_use_props.useProps("CheckboxIndicator", defaultProps, _props);
	const { classNames, className, style, styles, unstyled, vars, icon: Icon, indeterminate, radius, color, iconColor, autoContrast, checked, mod, variant, disabled, attributes, ...others } = props;
	const getStyles = require_use_styles.useStyles({
		name: "CheckboxIndicator",
		classes: require_CheckboxIndicator_module.default,
		props,
		className,
		style,
		classNames,
		styles,
		unstyled,
		attributes,
		vars,
		varsResolver,
		rootSelector: "indicator"
	});
	const ctx = (0, react.use)(require_CheckboxCard.CheckboxCardContext);
	const _checked = typeof checked === "boolean" || typeof indeterminate === "boolean" ? checked || indeterminate : ctx?.checked || false;
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(require_Box.Box, {
		...getStyles("indicator", { variant }),
		variant,
		mod: [{
			checked: _checked,
			disabled
		}, mod],
		...others,
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(Icon, {
			indeterminate,
			...getStyles("icon")
		})
	});
});
CheckboxIndicator.displayName = "@mantine/core/CheckboxIndicator";
CheckboxIndicator.classes = require_CheckboxIndicator_module.default;
CheckboxIndicator.varsResolver = varsResolver;
//#endregion
exports.CheckboxIndicator = CheckboxIndicator;
