// @mantine/core/esm/components/Tabs/TabsList/TabsList.mjs
"use client";
import { useProps } from "../../../core/MantineProvider/use-props/use-props.mjs";
import { factory } from "../../../core/factory/factory.mjs";
import { Box } from "../../../core/Box/Box.mjs";
import { useTabsContext } from "../Tabs.context.mjs";
import Tabs_module_default from "../Tabs.module.mjs";
import { jsx } from "react/jsx-runtime";
//#region packages/@mantine/core/src/components/Tabs/TabsList/TabsList.tsx
const TabsList = factory((_props) => {
	const props = useProps("TabsList", null, _props);
	const { children, className, grow, justify, classNames, styles, style, mod, ...others } = props;
	const ctx = useTabsContext();
	return /* @__PURE__ */ jsx(Box, {
		...ctx.getStyles("list", {
			className,
			style,
			classNames,
			styles,
			props,
			variant: ctx.variant
		}),
		role: "tablist",
		variant: ctx.variant,
		mod: [{
			grow,
			orientation: ctx.orientation,
			placement: ctx.orientation === "vertical" && ctx.placement,
			inverted: ctx.inverted
		}, mod],
		"aria-orientation": ctx.orientation,
		__vars: { "--tabs-justify": justify },
		...others,
		children
	});
});
TabsList.classes = Tabs_module_default;
TabsList.displayName = "@mantine/core/TabsList";
//#endregion
export { TabsList };
