// @mantine/core/cjs/components/HoverCard/HoverCardGroup/HoverCardGroup.cjs
"use client";
require("../../../_virtual/_rolldown/runtime.cjs");
const require_use_props = require("../../../core/MantineProvider/use-props/use-props.cjs");
let react = require("react");
let react_jsx_runtime = require("react/jsx-runtime");
let _floating_ui_react = require("@floating-ui/react");
//#region packages/@mantine/core/src/components/HoverCard/HoverCardGroup/HoverCardGroup.tsx
const HoverCardGroupContext = (0, react.createContext)({ withinGroup: false });
const defaultProps = {
	openDelay: 0,
	closeDelay: 0
};
function HoverCardGroup(props) {
	const { openDelay, closeDelay, children } = require_use_props.useProps("HoverCardGroup", defaultProps, props);
	return /* @__PURE__ */ (0, react_jsx_runtime.jsx)(HoverCardGroupContext, {
		value: { withinGroup: true },
		children: /* @__PURE__ */ (0, react_jsx_runtime.jsx)(_floating_ui_react.FloatingDelayGroup, {
			delay: {
				open: openDelay,
				close: closeDelay
			},
			children
		})
	});
}
HoverCardGroup.displayName = "@mantine/core/HoverCardGroup";
HoverCardGroup.extend = (c) => c;
//#endregion
exports.HoverCardGroup = HoverCardGroup;
exports.HoverCardGroupContext = HoverCardGroupContext;
